// @builder.io/sdk-react/lib/edge/blocks-0269007e.js
import { jsx as T, Fragment as X, jsxs as it } from "react/jsx-runtime";
import { createContext as Cn, useEffect as ye, useState as Ce, useRef as On, useContext as Mr } from "react";
import { BuilderContext as Xr } from "./blocks-exports.mjs";
import { isEditing as Lt, isBrowser as Ft, getUserAttributes as jn, checkIsDefined as st, logger as Hr, TARGET as lt, isPreviewing as An } from "./server-entry-fb5f2b80.js";
import { getClassPropName as In } from "./get-class-prop-name-770bdff2.js";
import { DynamicRenderer as Tn } from "./dynamic-renderer-8b343325.js";
const Pn = Cn({ registeredComponents: {} }), bn = (e, t, n) => {
  const r = String.prototype.split.call(t, /[,[\].]+?/).filter(Boolean).reduce((o, s) => o != null ? o[s] : o, e);
  return r === void 0 || r === e ? n : r;
}, Jr = ({
  builder: e,
  context: t,
  event: n,
  state: r
}) => Object.entries({
  state: r,
  Builder: e,
  // legacy
  builder: e,
  context: t,
  event: n
}), Nn = () => ({
  isEditing: Lt(),
  isBrowser: Ft(),
  isServer: !Ft(),
  getUserAttributes: () => jn()
}), Rn = (e, {
  isExpression: t = !0
}) => /* we disable this for cases where we definitely don't want a return */ t && !(e.includes(";") || e.includes(" return ") || e.trim().startsWith("return ")) ? `return (${e});` : e;
function Xt({
  rootState: e,
  localState: t,
  rootSetState: n
}) {
  return new Proxy(e, {
    get: (r, o) => {
      if (t && o in t)
        return t[o];
      const s = r[o];
      return typeof s == "object" && s !== null ? Xt({
        rootState: s,
        localState: void 0,
        rootSetState: n ? (a) => {
          r[o] = a, n(r);
        } : void 0
      }) : s;
    },
    set: (r, o, s) => {
      if (t && o in t)
        throw new Error("Writing to local state is not allowed as it is read-only.");
      return r[o] = s, n == null || n(r), !0;
    }
  });
}
const Bn = ({
  code: e,
  builder: t,
  context: n,
  event: r,
  localState: o,
  rootSetState: s,
  rootState: a
}) => {
  const f = Jr({
    builder: t,
    context: n,
    event: r,
    state: Xt({
      rootState: a,
      localState: o,
      rootSetState: s
    })
  });
  return new Function(...f.map(([d]) => d), e)(...f.map(([, d]) => d));
};
var b, Mn = function(e) {
  function t(i) {
    return 48 > i ? i === 36 : 58 > i ? !0 : 65 > i ? !1 : 91 > i ? !0 : 97 > i ? i === 95 : 123 > i ? !0 : 170 <= i && En.test(String.fromCharCode(i));
  }
  function n(i) {
    return 65 > i ? i === 36 : 91 > i ? !0 : 97 > i ? i === 95 : 123 > i ? !0 : 170 <= i && Br.test(String.fromCharCode(i));
  }
  function r(i, l) {
    for (var u = g, p = 1, E = 0; ; ) {
      ot.lastIndex = E;
      var L = ot.exec(u);
      if (L && L.index < i)
        ++p, E = L.index + L[0].length;
      else
        break;
    }
    throw u = {
      line: p,
      ab: i - E
    }, l += " (" + u.line + ":" + u.ab + ")", l = new SyntaxError(l), l.j = i, l.X = u, l.o = c, l;
  }
  function o(i) {
    i = i.split(" ");
    for (var l = /* @__PURE__ */ Object.create(null), u = 0; u < i.length; u++)
      l[i[u]] = !0;
    return function(p) {
      return l[p] || !1;
    };
  }
  function s() {
    this.line = de, this.ab = c - re;
  }
  function a(i, l) {
    ve = c, C.C && (kt = new s()), S = i, d(), Y = l, Re = i.m;
  }
  function f() {
    for (var i = c, l = C.va && C.C && new s(), u = g.charCodeAt(c += 2); c < me && u !== 10 && u !== 13 && u !== 8232 && u !== 8233; )
      ++c, u = g.charCodeAt(c);
    C.va && C.va(!1, g.slice(i + 2, c), i, c, l, C.C && new s());
  }
  function d() {
    for (; c < me; ) {
      var i = g.charCodeAt(c);
      if (i === 32)
        ++c;
      else if (i === 13)
        ++c, i = g.charCodeAt(c), i === 10 && ++c, C.C && (++de, re = c);
      else if (i === 10 || i === 8232 || i === 8233)
        ++c, C.C && (++de, re = c);
      else if (8 < i && 14 > i)
        ++c;
      else if (i === 47)
        if (i = g.charCodeAt(c + 1), i === 42) {
          i = void 0;
          var l = C.va && C.C && new s(), u = c, p = g.indexOf("*/", c += 2);
          if (p === -1 && r(c - 2, "Unterminated comment"), c = p + 2, C.C)
            for (ot.lastIndex = u; (i = ot.exec(g)) && i.index < c; )
              ++de, re = i.index + i[0].length;
          C.va && C.va(!0, g.slice(u + 2, p), u, c, l, C.C && new s());
        } else if (i === 47)
          f();
        else
          break;
      else if (i === 160)
        ++c;
      else if (5760 <= i && Sn.test(String.fromCharCode(i)))
        ++c;
      else
        break;
    }
  }
  function v(i) {
    switch (i) {
      case 46:
        i = g.charCodeAt(c + 1), 48 <= i && 57 >= i ? P(!0) : (++c, a(jr));
        return;
      case 40:
        return ++c, a(ne);
      case 41:
        return ++c, a(q);
      case 59:
        return ++c, a(oe);
      case 44:
        return ++c, a(fe);
      case 91:
        return ++c, a(Rt);
      case 93:
        return ++c, a(Bt);
      case 123:
        return ++c, a(De);
      case 125:
        return ++c, a(xe);
      case 58:
        return ++c, a(_e);
      case 63:
        return ++c, a(Ar);
      case 48:
        if (i = g.charCodeAt(c + 1), i === 120 || i === 88) {
          c += 2, i = V(16), i === null && r(U + 2, "Expected hexadecimal number"), n(g.charCodeAt(c)) && r(c, "Identifier directly after number"), a(Me, i);
          return;
        }
      case 49:
      case 50:
      case 51:
      case 52:
      case 53:
      case 54:
      case 55:
      case 56:
      case 57:
        return P(!1);
      case 34:
      case 39:
        c++;
        for (var l = ""; ; ) {
          c >= me && r(U, "Unterminated string constant");
          var u = g.charCodeAt(c);
          if (u === i) {
            ++c, a(et, l);
            break;
          }
          if (u === 92) {
            u = g.charCodeAt(++c);
            var p = /^[0-7]+/.exec(g.slice(c, c + 3));
            for (p && (p = p[0]); p && 255 < parseInt(p, 8); )
              p = p.slice(0, -1);
            if (p === "0" && (p = null), ++c, p)
              z && r(c - 2, "Octal literal in strict mode"), l += String.fromCharCode(parseInt(p, 8)), c += p.length - 1;
            else
              switch (u) {
                case 110:
                  l += `
`;
                  break;
                case 114:
                  l += "\r";
                  break;
                case 120:
                  l += String.fromCharCode(O(2));
                  break;
                case 117:
                  l += String.fromCharCode(O(4));
                  break;
                case 85:
                  l += String.fromCharCode(O(8));
                  break;
                case 116:
                  l += "	";
                  break;
                case 98:
                  l += "\b";
                  break;
                case 118:
                  l += "\v";
                  break;
                case 102:
                  l += "\f";
                  break;
                case 48:
                  l += "\0";
                  break;
                case 13:
                  g.charCodeAt(c) === 10 && ++c;
                case 10:
                  C.C && (re = c, ++de);
                  break;
                default:
                  l += String.fromCharCode(u);
              }
          } else
            u !== 13 && u !== 10 && u !== 8232 && u !== 8233 || r(U, "Unterminated string constant"), l += String.fromCharCode(u), ++c;
        }
        return;
      case 47:
        i = g.charCodeAt(c + 1), Re ? (++c, W()) : i === 61 ? A(ge, 2) : A(Ir, 1);
        return;
      case 37:
      case 42:
        g.charCodeAt(c + 1) === 61 ? A(ge, 2) : A(vn, 1);
        return;
      case 124:
      case 38:
        l = g.charCodeAt(c + 1), l === i ? A(i === 124 ? br : Nr, 2) : l === 61 ? A(ge, 2) : A(i === 124 ? fn : hn, 1);
        return;
      case 94:
        g.charCodeAt(c + 1) === 61 ? A(ge, 2) : A(pn, 1);
        return;
      case 43:
      case 45:
        l = g.charCodeAt(c + 1), l === i ? l === 45 && g.charCodeAt(c + 2) === 62 && nt.test(g.slice(ce, c)) ? (c += 3, f(), d(), I()) : A(cn, 2) : l === 61 ? A(ge, 2) : A(mn, 1);
        return;
      case 60:
      case 62:
        l = g.charCodeAt(c + 1), u = 1, l === i ? (u = i === 62 && g.charCodeAt(c + 2) === 62 ? 3 : 2, g.charCodeAt(c + u) === 61 ? A(ge, u + 1) : A(yn, u)) : l === 33 && i === 60 && g.charCodeAt(c + 2) === 45 && g.charCodeAt(c + 3) === 45 ? (c += 4, f(), d(), I()) : (l === 61 && (u = g.charCodeAt(c + 2) === 61 ? 3 : 2), A(gn, u));
        return;
      case 61:
      case 33:
        g.charCodeAt(c + 1) === 61 ? A(dn, g.charCodeAt(c + 2) === 61 ? 3 : 2) : A(i === 61 ? Tr : Pr, 1);
        return;
      case 126:
        return A(Pr, 1);
    }
    return !1;
  }
  function I(i) {
    if (i ? c = U + 1 : U = c, C.C && (Et = new s()), i)
      return W();
    if (c >= me)
      return a(jt);
    if (i = g.charCodeAt(c), n(i) || i === 92)
      return _();
    if (v(i) === !1) {
      if (i = String.fromCharCode(i), i === "\\" || Br.test(i))
        return _();
      r(c, "Unexpected character '" + i + "'");
    }
  }
  function A(i, l) {
    var u = g.slice(c, c + l);
    c += l, a(i, u);
  }
  function W() {
    for (var i, l, u = c; ; ) {
      c >= me && r(u, "Unterminated regexp");
      var p = g.charAt(c);
      if (nt.test(p) && r(u, "Unterminated regexp"), i)
        i = !1;
      else {
        if (p === "[")
          l = !0;
        else if (p === "]" && l)
          l = !1;
        else if (p === "/" && !l)
          break;
        i = p === "\\";
      }
      ++c;
    }
    i = g.slice(u, c), ++c, (l = R()) && !/^[gmi]*$/.test(l) && r(u, "Invalid regexp flag");
    try {
      var E = new RegExp(i, l);
    } catch (L) {
      throw L instanceof SyntaxError && r(u, L.message), L;
    }
    a(ar, E);
  }
  function V(i, l) {
    for (var u = c, p = 0, E = l === void 0 ? 1 / 0 : l, L = 0; L < E; ++L) {
      var H = g.charCodeAt(c);
      if (H = 97 <= H ? H - 97 + 10 : 65 <= H ? H - 65 + 10 : 48 <= H && 57 >= H ? H - 48 : 1 / 0, H >= i)
        break;
      ++c, p = p * i + H;
    }
    return c === u || l !== void 0 && c - u !== l ? null : p;
  }
  function P(i) {
    var l = c, u = !1, p = g.charCodeAt(c) === 48;
    i || V(10) !== null || r(l, "Invalid number"), g.charCodeAt(c) === 46 && (++c, V(10), u = !0), i = g.charCodeAt(c), (i === 69 || i === 101) && (i = g.charCodeAt(++c), i !== 43 && i !== 45 || ++c, V(10) === null && r(l, "Invalid number"), u = !0), n(g.charCodeAt(c)) && r(c, "Identifier directly after number"), i = g.slice(l, c);
    var E;
    u ? E = parseFloat(i) : p && i.length !== 1 ? /[89]/.test(i) || z ? r(l, "Invalid number") : E = parseInt(i, 8) : E = parseInt(i, 10), a(Me, E);
  }
  function O(i) {
    return i = V(16, i), i === null && r(U, "Bad character escape sequence"), i;
  }
  function R() {
    Se = !1;
    for (var i, l = !0, u = c; ; ) {
      var p = g.charCodeAt(c);
      if (t(p))
        Se && (i += g.charAt(c)), ++c;
      else if (p === 92) {
        Se || (i = g.slice(u, c)), Se = !0, g.charCodeAt(++c) !== 117 && r(c, "Expecting Unicode escape sequence \\uXXXX"), ++c, p = O(4);
        var E = String.fromCharCode(p);
        E || r(c - 1, "Invalid Unicode escape"), (l ? n(p) : t(p)) || r(c - 4, "Invalid Unicode escape"), i += E;
      } else
        break;
      l = !1;
    }
    return Se ? i : g.slice(u, c);
  }
  function _() {
    var i = R(), l = we;
    !Se && xn(i) && (l = un[i]), a(l, i);
  }
  function k() {
    Ct = U, ce = ve, Ot = kt, I();
  }
  function ue(i) {
    if (z = i, c = U, C.C)
      for (; c < re; )
        re = g.lastIndexOf(`
`, re - 2) + 1, --de;
    d(), I();
  }
  function Ie() {
    this.type = null, this.start = U, this.end = null;
  }
  function Te() {
    this.start = Et, this.end = null, St && (this.source = St);
  }
  function B() {
    var i = new Ie();
    return C.C && (i.X = new Te()), C.vb && (i.sourceFile = C.vb), C.Xa && (i.j = [U, 0]), i;
  }
  function Z(i) {
    var l = new Ie();
    return l.start = i.start, C.C && (l.X = new Te(), l.X.start = i.X.start), C.Xa && (l.j = [i.j[0], 0]), l;
  }
  function x(i, l) {
    return i.type = l, i.end = ce, C.C && (i.X.end = Ot), C.Xa && (i.j[1] = ce), i;
  }
  function dt(i) {
    return i.type === "ExpressionStatement" && i.la.type === "Literal" && i.la.value === "use strict";
  }
  function N(i) {
    return S === i ? (k(), !0) : !1;
  }
  function Ze() {
    return !C.fc && (S === jt || S === xe || nt.test(g.slice(ce, U)));
  }
  function he() {
    N(oe) || Ze() || ee();
  }
  function M(i) {
    S === i ? k() : ee();
  }
  function ee() {
    r(U, "Unexpected token");
  }
  function Qe(i) {
    i.type !== "Identifier" && i.type !== "MemberExpression" && r(i.start, "Assigning to rvalue"), z && i.type === "Identifier" && rt(i.name) && r(i.start, "Assigning to " + i.name + " in strict mode");
  }
  function Q() {
    (S === Ir || S === ge && Y === "/=") && I(!0);
    var i = S, l = B();
    switch (i) {
      case At:
      case ur:
        k();
        var u = i === At;
        N(oe) || Ze() ? l.label = null : S !== we ? ee() : (l.label = te(), he());
        for (var p = 0; p < D.length; ++p) {
          var E = D[p];
          if ((l.label === null || E.name === l.label.name) && (E.kind !== null && (u || E.kind === "loop") || l.label && u))
            break;
        }
        return p === D.length && r(l.start, "Unsyntactic " + i.l), x(l, u ? "BreakStatement" : "ContinueStatement");
      case cr:
        return k(), he(), x(l, "DebuggerStatement");
      case pr:
        return k(), D.push(Mt), l.body = Q(), D.pop(), M(Nt), l.test = Pe(), he(), x(l, "DoWhileStatement");
      case gr:
        return k(), D.push(Mt), M(ne), S === oe ? gt(l, null) : S === bt ? (i = B(), k(), ir(i, !0), x(i, "VariableDeclaration"), i.fa.length === 1 && N(tt) ? or(l, i) : gt(l, i)) : (i = $(!1, !0), N(tt) ? (Qe(i), or(l, i)) : gt(l, i));
      case Tt:
        return k(), wt(l, !0);
      case yr:
        return k(), l.test = Pe(), l.da = Q(), l.alternate = N(hr) ? Q() : null, x(l, "IfStatement");
      case mr:
        return Be || C.Ib || r(U, "'return' outside of function"), k(), N(oe) || Ze() ? l.K = null : (l.K = $(), he()), x(l, "ReturnStatement");
      case Pt:
        for (k(), l.Qb = Pe(), l.tb = [], M(De), D.push(kn); S !== xe; )
          S === It || S === fr ? (i = S === It, p && x(p, "SwitchCase"), l.tb.push(p = B()), p.da = [], k(), i ? p.test = $() : (u && r(Ct, "Multiple default clauses"), u = !0, p.test = null), M(_e)) : (p || ee(), p.da.push(Q()));
        return p && x(p, "SwitchCase"), k(), D.pop(), x(l, "SwitchStatement");
      case vr:
        return k(), nt.test(g.slice(ce, U)) && r(ce, "Illegal newline after throw"), l.K = $(), he(), x(l, "ThrowStatement");
      case wr:
        return k(), l.block = be(), l.Ea = null, S === lr && (i = B(), k(), M(ne), i.Ua = te(), z && rt(i.Ua.name) && r(i.Ua.start, "Binding " + i.Ua.name + " in strict mode"), M(q), i.body = be(), l.Ea = x(i, "CatchClause")), l.fb = N(dr) ? be() : null, l.Ea || l.fb || r(l.start, "Missing catch or finally clause"), x(l, "TryStatement");
      case bt:
        return k(), ir(l), he(), x(l, "VariableDeclaration");
      case Nt:
        return k(), l.test = Pe(), D.push(Mt), l.body = Q(), D.pop(), x(l, "WhileStatement");
      case xr:
        return z && r(U, "'with' in strict mode"), k(), l.object = Pe(), l.body = Q(), x(l, "WithStatement");
      case De:
        return be();
      case oe:
        return k(), x(l, "EmptyStatement");
      default:
        if (u = Y, E = $(), i === we && E.type === "Identifier" && N(_e)) {
          for (p = 0; p < D.length; ++p)
            D[p].name === u && r(E.start, "Label '" + u + "' is already declared");
          return D.push({
            name: u,
            kind: S.W ? "loop" : S === Pt ? "switch" : null
          }), l.body = Q(), D.pop(), l.label = E, x(l, "LabeledStatement");
        }
        return l.la = E, he(), x(l, "ExpressionStatement");
    }
  }
  function Pe() {
    M(ne);
    var i = $();
    return M(q), i;
  }
  function be(i) {
    var l = B(), u = !0, p = !1;
    for (l.body = [], M(De); !N(xe); ) {
      var E = Q();
      if (l.body.push(E), u && i && dt(E)) {
        var L = p;
        ue(p = !0);
      }
      u = !1;
    }
    return p && !L && ue(!1), x(l, "BlockStatement");
  }
  function gt(i, l) {
    return i.ua = l, M(oe), i.test = S === oe ? null : $(), M(oe), i.update = S === q ? null : $(), M(q), i.body = Q(), D.pop(), x(i, "ForStatement");
  }
  function or(i, l) {
    return i.left = l, i.right = $(), M(q), i.body = Q(), D.pop(), x(i, "ForInStatement");
  }
  function ir(i, l) {
    for (i.fa = [], i.kind = "var"; ; ) {
      var u = B();
      if (u.id = te(), z && rt(u.id.name) && r(u.id.start, "Binding " + u.id.name + " in strict mode"), u.ua = N(Tr) ? $(!0, l) : null, i.fa.push(x(u, "VariableDeclarator")), !N(fe))
        break;
    }
  }
  function $(i, l) {
    var u = yt(l);
    if (!i && S === fe) {
      for (i = Z(u), i.xb = [u]; N(fe); )
        i.xb.push(yt(l));
      return x(i, "SequenceExpression");
    }
    return u;
  }
  function yt(i) {
    var l = mt(vt(), -1, i);
    if (N(Ar)) {
      var u = Z(l);
      u.test = l, u.da = $(!0), M(_e), u.alternate = $(!0, i), l = x(u, "ConditionalExpression");
    }
    return S.Cb ? (u = Z(l), u.operator = Y, u.left = l, k(), u.right = yt(i), Qe(l), x(u, "AssignmentExpression")) : l;
  }
  function mt(i, l, u) {
    var p = S.L;
    if (p !== null && (!u || S !== tt) && p > l) {
      var E = Z(i);
      return E.left = i, E.operator = Y, i = S, k(), E.right = mt(vt(), p, u), p = x(E, i === br || i === Nr ? "LogicalExpression" : "BinaryExpression"), mt(p, l, u);
    }
    return i;
  }
  function vt() {
    if (S.prefix) {
      var i = B(), l = S.$b;
      return i.operator = Y, Re = i.prefix = !0, k(), i.K = vt(), l ? Qe(i.K) : z && i.operator === "delete" && i.K.type === "Identifier" && r(i.start, "Deleting local variable in strict mode"), x(i, l ? "UpdateExpression" : "UnaryExpression");
    }
    for (l = Ne(qe()); S.cc && !Ze(); )
      i = Z(l), i.operator = Y, i.prefix = !1, i.K = l, Qe(l), k(), l = x(i, "UpdateExpression");
    return l;
  }
  function Ne(i, l) {
    if (N(jr)) {
      var u = Z(i);
      return u.object = i, u.Wa = te(!0), u.bb = !1, Ne(x(u, "MemberExpression"), l);
    }
    return N(Rt) ? (u = Z(i), u.object = i, u.Wa = $(), u.bb = !0, M(Bt), Ne(x(u, "MemberExpression"), l)) : !l && N(ne) ? (u = Z(i), u.callee = i, u.arguments = xt(q, !1), Ne(x(u, "CallExpression"), l)) : i;
  }
  function qe() {
    switch (S) {
      case Er:
        var i = B();
        return k(), x(i, "ThisExpression");
      case we:
        return te();
      case Me:
      case et:
      case ar:
        return i = B(), i.value = Y, i.raw = g.slice(U, ve), k(), x(i, "Literal");
      case kr:
      case Cr:
      case Or:
        return i = B(), i.value = S.$a, i.raw = S.l, k(), x(i, "Literal");
      case ne:
        i = Et;
        var l = U;
        k();
        var u = $();
        return u.start = l, u.end = ve, C.C && (u.X.start = i, u.X.end = kt), C.Xa && (u.j = [l, ve]), M(q), u;
      case Rt:
        return i = B(), k(), i.elements = xt(Bt, !0, !0), x(i, "ArrayExpression");
      case De:
        for (i = B(), l = !0, u = !1, i.h = [], k(); !N(xe); ) {
          if (l)
            l = !1;
          else if (M(fe), C.sb && N(xe))
            break;
          var p = {
            key: S === Me || S === et ? qe() : te(!0)
          }, E = !1;
          if (N(_e)) {
            p.value = $(!0);
            var L = p.kind = "init";
          } else
            p.key.type !== "Identifier" || p.key.name !== "get" && p.key.name !== "set" ? ee() : (E = u = !0, L = p.kind = p.key.name, p.key = S === Me || S === et ? qe() : te(!0), S !== ne && ee(), p.value = wt(B(), !1));
          if (p.key.type === "Identifier" && (z || u))
            for (var H = 0; H < i.h.length; ++H) {
              var Ee = i.h[H];
              if (Ee.key.name === p.key.name) {
                var Dt = L === Ee.kind || E && Ee.kind === "init" || L === "init" && (Ee.kind === "get" || Ee.kind === "set");
                Dt && !z && L === "init" && Ee.kind === "init" && (Dt = !1), Dt && r(p.key.start, "Redefinition of property");
              }
            }
          i.h.push(p);
        }
        return x(i, "ObjectExpression");
      case Tt:
        return i = B(), k(), wt(i, !1);
      case Sr:
        return i = B(), k(), i.callee = Ne(qe(), !0), i.arguments = N(ne) ? xt(q, !1) : ln, x(i, "NewExpression");
    }
    ee();
  }
  function wt(i, l) {
    S === we ? i.id = te() : l ? ee() : i.id = null, i.oa = [];
    var u = !0;
    for (M(ne); !N(q); )
      u ? u = !1 : M(fe), i.oa.push(te());
    u = Be;
    var p = D;
    if (Be = !0, D = [], i.body = be(!0), Be = u, D = p, z || i.body.body.length && dt(i.body.body[0])) {
      for (u = i.id ? -1 : 0; u < i.oa.length; ++u)
        if (p = 0 > u ? i.id : i.oa[u], (Rr(p.name) || rt(p.name)) && r(p.start, "Defining '" + p.name + "' in strict mode"), 0 <= u)
          for (var E = 0; E < u; ++E)
            p.name === i.oa[E].name && r(p.start, "Argument name clash in strict mode");
    }
    return x(i, l ? "FunctionDeclaration" : "FunctionExpression");
  }
  function xt(i, l, u) {
    for (var p = [], E = !0; !N(i); ) {
      if (E)
        E = !1;
      else if (M(fe), l && C.sb && N(i))
        break;
      p.push(u && S === fe ? null : $(!0));
    }
    return p;
  }
  function te(i) {
    var l = B();
    return i && C.yb === "everywhere" && (i = !1), S === we ? (!i && (C.yb && wn(Y) || z && Rr(Y)) && g.slice(U, ve).indexOf("\\") === -1 && r(U, "The keyword '" + Y + "' is reserved"), l.name = Y) : i && S.l ? l.name = S.l : ee(), Re = !1, k(), x(l, "Identifier");
  }
  e.version = "0.5.0";
  var C, g = "", me, St;
  e.parse = function(i, l) {
    g = String(i), me = g.length, C = l || {};
    for (var u in sr)
      Object.prototype.hasOwnProperty.call(C, u) || (C[u] = sr[u]);
    for (St = C.sourceFile, de = 1, c = re = 0, Re = !0, d(), u = C.dc, Ct = ce = c, C.C && (Ot = new s()), Be = z = !1, D = [], I(), i = u || B(), l = !0, u || (i.body = []); S !== jt; )
      u = Q(), i.body.push(u), l && dt(u) && ue(!0), l = !1;
    return x(i, "Program");
  };
  var sr = {
    fc: !1,
    sb: !0,
    yb: !1,
    Ib: !1,
    C: !1,
    va: null,
    Xa: !1,
    dc: null,
    sourceFile: null,
    vb: null
  }, c = 0, U = 0, ve = 0, Et, kt, S, Y, Re, de, re, Ct = 0, ce = 0, Ot, Be, D, z, ln = [], Me = {
    type: "num"
  }, ar = {
    type: "regexp"
  }, et = {
    type: "string"
  }, we = {
    type: "name"
  }, jt = {
    type: "eof"
  }, At = {
    l: "break"
  }, It = {
    l: "case",
    m: !0
  }, lr = {
    l: "catch"
  }, ur = {
    l: "continue"
  }, cr = {
    l: "debugger"
  }, fr = {
    l: "default"
  }, pr = {
    l: "do",
    W: !0
  }, hr = {
    l: "else",
    m: !0
  }, dr = {
    l: "finally"
  }, gr = {
    l: "for",
    W: !0
  }, Tt = {
    l: "function"
  }, yr = {
    l: "if"
  }, mr = {
    l: "return",
    m: !0
  }, Pt = {
    l: "switch"
  }, vr = {
    l: "throw",
    m: !0
  }, wr = {
    l: "try"
  }, bt = {
    l: "var"
  }, Nt = {
    l: "while",
    W: !0
  }, xr = {
    l: "with"
  }, Sr = {
    l: "new",
    m: !0
  }, Er = {
    l: "this"
  }, kr = {
    l: "null",
    $a: null
  }, Cr = {
    l: "true",
    $a: !0
  }, Or = {
    l: "false",
    $a: !1
  }, tt = {
    l: "in",
    L: 7,
    m: !0
  }, un = {
    break: At,
    case: It,
    catch: lr,
    continue: ur,
    debugger: cr,
    default: fr,
    do: pr,
    else: hr,
    finally: dr,
    for: gr,
    function: Tt,
    if: yr,
    return: mr,
    switch: Pt,
    throw: vr,
    try: wr,
    var: bt,
    while: Nt,
    with: xr,
    null: kr,
    true: Cr,
    false: Or,
    new: Sr,
    in: tt,
    instanceof: {
      l: "instanceof",
      L: 7,
      m: !0
    },
    this: Er,
    typeof: {
      l: "typeof",
      prefix: !0,
      m: !0
    },
    void: {
      l: "void",
      prefix: !0,
      m: !0
    },
    delete: {
      l: "delete",
      prefix: !0,
      m: !0
    }
  }, Rt = {
    type: "[",
    m: !0
  }, Bt = {
    type: "]"
  }, De = {
    type: "{",
    m: !0
  }, xe = {
    type: "}"
  }, ne = {
    type: "(",
    m: !0
  }, q = {
    type: ")"
  }, fe = {
    type: ",",
    m: !0
  }, oe = {
    type: ";",
    m: !0
  }, _e = {
    type: ":",
    m: !0
  }, jr = {
    type: "."
  }, Ar = {
    type: "?",
    m: !0
  }, Ir = {
    L: 10,
    m: !0
  }, Tr = {
    Cb: !0,
    m: !0
  }, ge = {
    Cb: !0,
    m: !0
  }, cn = {
    cc: !0,
    prefix: !0,
    $b: !0
  }, Pr = {
    prefix: !0,
    m: !0
  }, br = {
    L: 1,
    m: !0
  }, Nr = {
    L: 2,
    m: !0
  }, fn = {
    L: 3,
    m: !0
  }, pn = {
    L: 4,
    m: !0
  }, hn = {
    L: 5,
    m: !0
  }, dn = {
    L: 6,
    m: !0
  }, gn = {
    L: 7,
    m: !0
  }, yn = {
    L: 8,
    m: !0
  }, mn = {
    L: 9,
    prefix: !0,
    m: !0
  }, vn = {
    L: 10,
    m: !0
  }, wn = o("class enum extends super const export import"), Rr = o("implements interface let package private protected public static yield"), rt = o("eval arguments"), xn = o("break case catch continue debugger default do else finally for function if return switch throw try var while with null true false instanceof typeof void delete new in this"), Sn = /[\u1680\u180e\u2000-\u200a\u202f\u205f\u3000\ufeff]/, Br = RegExp("[ªµºÀ-ÖØ-öø-ˁˆ-ˑˠ-ˤˬˮͰ-ʹͶͷͺ-ͽΆΈ-ΊΌΎ-ΡΣ-ϵϷ-ҁҊ-ԧԱ-Ֆՙա-ևא-תװ-ײؠ-يٮٯٱ-ۓەۥۦۮۯۺ-ۼۿܐܒ-ܯݍ-ޥޱߊ-ߪߴߵߺࠀ-ࠕࠚࠤࠨࡀ-ࡘࢠࢢ-ࢬऄ-हऽॐक़-ॡॱ-ॷॹ-ॿঅ-ঌএঐও-নপ-রলশ-হঽৎড়ঢ়য়-ৡৰৱਅ-ਊਏਐਓ-ਨਪ-ਰਲਲ਼ਵਸ਼ਸਹਖ਼-ੜਫ਼ੲ-ੴઅ-ઍએ-ઑઓ-નપ-રલળવ-હઽૐૠૡଅ-ଌଏଐଓ-ନପ-ରଲଳଵ-ହଽଡ଼ଢ଼ୟ-ୡୱஃஅ-ஊஎ-ஐஒ-கஙசஜஞடணதந-பம-ஹௐఅ-ఌఎ-ఐఒ-నప-ళవ-హఽౘౙౠౡಅ-ಌಎ-ಐಒ-ನಪ-ಳವ-ಹಽೞೠೡೱೲഅ-ഌഎ-ഐഒ-ഺഽൎൠൡൺ-ൿඅ-ඖක-නඳ-රලව-ෆก-ะาำเ-ๆກຂຄງຈຊຍດ-ທນ-ຟມ-ຣລວສຫອ-ະາຳຽເ-ໄໆໜ-ໟༀཀ-ཇཉ-ཬྈ-ྌက-ဪဿၐ-ၕၚ-ၝၡၥၦၮ-ၰၵ-ႁႎႠ-ჅჇჍა-ჺჼ-ቈቊ-ቍቐ-ቖቘቚ-ቝበ-ኈኊ-ኍነ-ኰኲ-ኵኸ-ኾዀዂ-ዅወ-ዖዘ-ጐጒ-ጕጘ-ፚᎀ-ᎏᎠ-Ᏼᐁ-ᙬᙯ-ᙿᚁ-ᚚᚠ-ᛪᛮ-ᛰᜀ-ᜌᜎ-ᜑᜠ-ᜱᝀ-ᝑᝠ-ᝬᝮ-ᝰក-ឳៗៜᠠ-ᡷᢀ-ᢨᢪᢰ-ᣵᤀ-ᤜᥐ-ᥭᥰ-ᥴᦀ-ᦫᧁ-ᧇᨀ-ᨖᨠ-ᩔᪧᬅ-ᬳᭅ-ᭋᮃ-ᮠᮮᮯᮺ-ᯥᰀ-ᰣᱍ-ᱏᱚ-ᱽᳩ-ᳬᳮ-ᳱᳵᳶᴀ-ᶿḀ-ἕἘ-Ἕἠ-ὅὈ-Ὅὐ-ὗὙὛὝὟ-ώᾀ-ᾴᾶ-ᾼιῂ-ῄῆ-ῌῐ-ΐῖ-Ίῠ-Ῥῲ-ῴῶ-ῼⁱⁿₐ-ₜℂℇℊ-ℓℕℙ-ℝℤΩℨK-ℭℯ-ℹℼ-ℿⅅ-ⅉⅎⅠ-ↈⰀ-Ⱞⰰ-ⱞⱠ-ⳤⳫ-ⳮⳲⳳⴀ-ⴥⴧⴭⴰ-ⵧⵯⶀ-ⶖⶠ-ⶦⶨ-ⶮⶰ-ⶶⶸ-ⶾⷀ-ⷆⷈ-ⷎⷐ-ⷖⷘ-ⷞⸯ々-〇〡-〩〱-〵〸-〼ぁ-ゖゝ-ゟァ-ヺー-ヿㄅ-ㄭㄱ-ㆎㆠ-ㆺㇰ-ㇿ㐀-䶵一-鿌ꀀ-ꒌꓐ-ꓽꔀ-ꘌꘐ-ꘟꘪꘫꙀ-ꙮꙿ-ꚗꚠ-ꛯꜗ-ꜟꜢ-ꞈꞋ-ꞎꞐ-ꞓꞠ-Ɦꟸ-ꠁꠃ-ꠅꠇ-ꠊꠌ-ꠢꡀ-ꡳꢂ-ꢳꣲ-ꣷꣻꤊ-ꤥꤰ-ꥆꥠ-ꥼꦄ-ꦲꧏꨀ-ꨨꩀ-ꩂꩄ-ꩋꩠ-ꩶꩺꪀ-ꪯꪱꪵꪶꪹ-ꪽꫀꫂꫛ-ꫝꫠ-ꫪꫲ-ꫴꬁ-ꬆꬉ-ꬎꬑ-ꬖꬠ-ꬦꬨ-ꬮꯀ-ꯢ가-힣ힰ-ퟆퟋ-ퟻ豈-舘並-龎ﬀ-ﬆﬓ-ﬗיִײַ-ﬨשׁ-זּטּ-לּמּנּסּףּפּצּ-ﮱﯓ-ﴽﵐ-ﶏﶒ-ﷇﷰ-ﷻﹰ-ﹴﹶ-ﻼＡ-Ｚａ-ｚｦ-ﾾￂ-ￇￊ-ￏￒ-ￗￚ-ￜ]"), En = RegExp("[ªµºÀ-ÖØ-öø-ˁˆ-ˑˠ-ˤˬˮͰ-ʹͶͷͺ-ͽΆΈ-ΊΌΎ-ΡΣ-ϵϷ-ҁҊ-ԧԱ-Ֆՙա-ևא-תװ-ײؠ-يٮٯٱ-ۓەۥۦۮۯۺ-ۼۿܐܒ-ܯݍ-ޥޱߊ-ߪߴߵߺࠀ-ࠕࠚࠤࠨࡀ-ࡘࢠࢢ-ࢬऄ-हऽॐक़-ॡॱ-ॷॹ-ॿঅ-ঌএঐও-নপ-রলশ-হঽৎড়ঢ়য়-ৡৰৱਅ-ਊਏਐਓ-ਨਪ-ਰਲਲ਼ਵਸ਼ਸਹਖ਼-ੜਫ਼ੲ-ੴઅ-ઍએ-ઑઓ-નપ-રલળવ-હઽૐૠૡଅ-ଌଏଐଓ-ନପ-ରଲଳଵ-ହଽଡ଼ଢ଼ୟ-ୡୱஃஅ-ஊஎ-ஐஒ-கஙசஜஞடணதந-பம-ஹௐఅ-ఌఎ-ఐఒ-నప-ళవ-హఽౘౙౠౡಅ-ಌಎ-ಐಒ-ನಪ-ಳವ-ಹಽೞೠೡೱೲഅ-ഌഎ-ഐഒ-ഺഽൎൠൡൺ-ൿඅ-ඖක-නඳ-රලව-ෆก-ะาำเ-ๆກຂຄງຈຊຍດ-ທນ-ຟມ-ຣລວສຫອ-ະາຳຽເ-ໄໆໜ-ໟༀཀ-ཇཉ-ཬྈ-ྌက-ဪဿၐ-ၕၚ-ၝၡၥၦၮ-ၰၵ-ႁႎႠ-ჅჇჍა-ჺჼ-ቈቊ-ቍቐ-ቖቘቚ-ቝበ-ኈኊ-ኍነ-ኰኲ-ኵኸ-ኾዀዂ-ዅወ-ዖዘ-ጐጒ-ጕጘ-ፚᎀ-ᎏᎠ-Ᏼᐁ-ᙬᙯ-ᙿᚁ-ᚚᚠ-ᛪᛮ-ᛰᜀ-ᜌᜎ-ᜑᜠ-ᜱᝀ-ᝑᝠ-ᝬᝮ-ᝰក-ឳៗៜᠠ-ᡷᢀ-ᢨᢪᢰ-ᣵᤀ-ᤜᥐ-ᥭᥰ-ᥴᦀ-ᦫᧁ-ᧇᨀ-ᨖᨠ-ᩔᪧᬅ-ᬳᭅ-ᭋᮃ-ᮠᮮᮯᮺ-ᯥᰀ-ᰣᱍ-ᱏᱚ-ᱽᳩ-ᳬᳮ-ᳱᳵᳶᴀ-ᶿḀ-ἕἘ-Ἕἠ-ὅὈ-Ὅὐ-ὗὙὛὝὟ-ώᾀ-ᾴᾶ-ᾼιῂ-ῄῆ-ῌῐ-ΐῖ-Ίῠ-Ῥῲ-ῴῶ-ῼⁱⁿₐ-ₜℂℇℊ-ℓℕℙ-ℝℤΩℨK-ℭℯ-ℹℼ-ℿⅅ-ⅉⅎⅠ-ↈⰀ-Ⱞⰰ-ⱞⱠ-ⳤⳫ-ⳮⳲⳳⴀ-ⴥⴧⴭⴰ-ⵧⵯⶀ-ⶖⶠ-ⶦⶨ-ⶮⶰ-ⶶⶸ-ⶾⷀ-ⷆⷈ-ⷎⷐ-ⷖⷘ-ⷞⸯ々-〇〡-〩〱-〵〸-〼ぁ-ゖゝ-ゟァ-ヺー-ヿㄅ-ㄭㄱ-ㆎㆠ-ㆺㇰ-ㇿ㐀-䶵一-鿌ꀀ-ꒌꓐ-ꓽꔀ-ꘌꘐ-ꘟꘪꘫꙀ-ꙮꙿ-ꚗꚠ-ꛯꜗ-ꜟꜢ-ꞈꞋ-ꞎꞐ-ꞓꞠ-Ɦꟸ-ꠁꠃ-ꠅꠇ-ꠊꠌ-ꠢꡀ-ꡳꢂ-ꢳꣲ-ꣷꣻꤊ-ꤥꤰ-ꥆꥠ-ꥼꦄ-ꦲꧏꨀ-ꨨꩀ-ꩂꩄ-ꩋꩠ-ꩶꩺꪀ-ꪯꪱꪵꪶꪹ-ꪽꫀꫂꫛ-ꫝꫠ-ꫪꫲ-ꫴꬁ-ꬆꬉ-ꬎꬑ-ꬖꬠ-ꬦꬨ-ꬮꯀ-ꯢ가-힣ힰ-ퟆퟋ-ퟻ豈-舘並-龎ﬀ-ﬆﬓ-ﬗיִײַ-ﬨשׁ-זּטּ-לּמּנּסּףּפּצּ-ﮱﯓ-ﴽﵐ-ﶏﶒ-ﷇﷰ-ﷻﹰ-ﹴﹶ-ﻼＡ-Ｚａ-ｚｦ-ﾾￂ-ￇￊ-ￏￒ-ￗￚ-ￜ̀-ͯ҃-֑҇-ׇֽֿׁׂׅׄؐ-ؚؠ-ىٲ-ۓۧ-ۨۻ-ۼܰ-݊ࠀ-ࠔࠛ-ࠣࠥ-ࠧࠩ-࠭ࡀ-ࡗࣤ-ࣾऀ-ःऺ-़ा-ॏ॑-ॗॢ-ॣ०-९ঁ-ঃ়া-ৄেৈৗয়-ৠਁ-ਃ਼ਾ-ੂੇੈੋ-੍ੑ੦-ੱੵઁ-ઃ઼ા-ૅે-ૉો-્ૢ-ૣ૦-૯ଁ-ଃ଼ା-ୄେୈୋ-୍ୖୗୟ-ୠ୦-୯ஂா-ூெ-ைொ-்ௗ௦-௯ఁ-ఃె-ైొ-్ౕౖౢ-ౣ౦-౯ಂಃ಼ಾ-ೄೆ-ೈೊ-್ೕೖೢ-ೣ೦-೯ംഃെ-ൈൗൢ-ൣ൦-൯ංඃ්ා-ුූෘ-ෟෲෳิ-ฺเ-ๅ๐-๙ິ-ູ່-ໍ໐-໙༘༙༠-༩༹༵༷ཁ-ཇཱ-྄྆-྇ྍ-ྗྙ-ྼ࿆က-ဩ၀-၉ၧ-ၭၱ-ၴႂ-ႍႏ-ႝ፝-፟ᜎ-ᜐᜠ-ᜰᝀ-ᝐᝲᝳក-ឲ៝០-៩᠋-᠍᠐-᠙ᤠ-ᤫᤰ-᤻ᥑ-ᥭᦰ-ᧀᧈ-ᧉ᧐-᧙ᨀ-ᨕᨠ-ᩓ᩠-᩿᩼-᪉᪐-᪙ᭆ-ᭋ᭐-᭙᭫-᭳᮰-᮹᯦-᯳ᰀ-ᰢ᱀-᱉ᱛ-ᱽ᳐-᳒ᴀ-ᶾḁ-ἕ‌‍‿⁀⁔⃐-⃥⃜⃡-⃰ⶁ-ⶖⷠ-ⷿ〡-〨゙゚Ꙁ-ꙭꙴ-꙽ꚟ꛰-꛱ꟸ-ꠀ꠆ꠋꠣ-ꠧꢀ-ꢁꢴ-꣄꣐-꣙ꣳ-ꣷ꤀-꤉ꤦ-꤭ꤰ-ꥅꦀ-ꦃ꦳-꧀ꨀ-ꨧꩀ-ꩁꩌ-ꩍ꩐-꩙ꩻꫠ-ꫩꫲ-ꫳꯀ-ꯡ꯬꯭꯰-꯹ﬠ-ﬨ︀-️︠-︦︳︴﹍-﹏０-９＿]"), nt = /[\n\r\u2028\u2029]/, ot = /\r\n|[\n\r\u2028\u2029]/g, Se, Mt = {
    kind: "loop"
  }, kn = {
    kind: "switch"
  };
}, Wt = Wt = typeof globalThis == "undefined" ? void 0 : globalThis;
Mn(Wt.j || (Wt.j = {}));
function h(e, t) {
  typeof e == "string" && (e = Oe(e, "code"));
  var n = e.constructor;
  this.ya = function() {
    return new n({
      options: {}
    });
  };
  var r = this.ya(), o;
  for (o in e)
    r[o] = o === "body" ? e[o].slice() : e[o];
  this.ra = r, this.ca = [], this.qb = t, this.za = !1, this.ba = [], this.Ya = 0, this.rb = /* @__PURE__ */ Object.create(null), e = /^step([A-Z]\w*)$/;
  var s, a;
  for (a in this)
    typeof this[a] == "function" && (s = a.match(e)) && (this.rb[s[1]] = this[a].bind(this));
  this.N = $t(this, this.ra, null), this.Na = this.N.object, this.ra = Oe(this.ba.join(`
`), "polyfills"), this.ba = void 0, ut(this.ra), s = new m(this.ra, this.N), s.done = !1, this.o = [s], this.Eb(), this.value = void 0, this.ra = r, s = new m(this.ra, this.N), s.done = !1, this.o.length = 0, this.o[0] = s;
}
var Dr = {
  C: !0,
  kc: 5
}, Ht = {
  configurable: !0,
  enumerable: !0,
  writable: !1
}, y = {
  configurable: !0,
  enumerable: !1,
  writable: !0
}, K = {
  configurable: !0,
  enumerable: !1,
  writable: !1
}, Le = {
  configurable: !1,
  enumerable: !1,
  writable: !1
}, _r = {
  configurable: !1,
  enumerable: !0,
  writable: !0
}, Kr = {
  STEP_ERROR: !0
}, Ae = {
  SCOPE_REFERENCE: !0
}, Je = {
  VALUE_IN_DESCRIPTOR: !0
}, ke = {
  REGEXP_TIMEOUT: !0
}, Ur = [], se = null, Ve = null, pe = typeof globalThis == "undefined" ? void 0 : globalThis, Dn = ["onmessage = function(e) {", "var result;", "var data = e.data;", "switch (data[0]) {", "case 'split':", "result = data[1].split(data[2], data[3]);", "break;", "case 'match':", "result = data[1].match(data[2]);", "break;", "case 'search':", "result = data[1].search(data[2]);", "break;", "case 'replace':", "result = data[1].replace(data[2], data[3]);", "break;", "case 'exec':", "var regexp = data[1];", "regexp.lastIndex = data[2];", "result = [regexp.exec(data[3]), data[1].lastIndex];", "break;", "default:", "throw Error('Unknown RegExp operation: ' + data[0]);", "}", "postMessage(result);", "close();", "};"];
function Jt(e) {
  var t = e >>> 0;
  return t === Number(e) ? t : NaN;
}
function Xe(e) {
  var t = e >>> 0;
  return String(t) === String(e) && t !== 4294967295 ? t : NaN;
}
function ut(e, t, n) {
  t ? e.start = t : delete e.start, n ? e.end = n : delete e.end;
  for (var r in e)
    if (r !== "loc" && e.hasOwnProperty(r)) {
      var o = e[r];
      o && typeof o == "object" && ut(o, t, n);
    }
}
h.prototype.REGEXP_MODE = 2;
h.prototype.REGEXP_THREAD_TIMEOUT = 1e3;
h.prototype.POLYFILL_TIMEOUT = 1e3;
b = h.prototype;
b.P = !1;
b.Ka = !1;
b.Kb = 0;
b.ic = 0;
function Oe(e, t) {
  var n = {}, r;
  for (r in Dr)
    n[r] = Dr[r];
  return n.sourceFile = t, pe.j.parse(e, n);
}
b.Jb = function(e) {
  var t = this.o[0];
  if (!t || t.node.type !== "Program")
    throw Error("Expecting original AST to start with a Program node");
  if (typeof e == "string" && (e = Oe(e, "appendCode" + this.Kb++)), !e || e.type !== "Program")
    throw Error("Expecting new AST to start with a Program node");
  Ke(this, e, t.scope), Array.prototype.push.apply(t.node.body, e.body), t.node.body.jb = null, t.done = !1;
};
b.lb = function() {
  var e = this.o, t;
  do {
    var n = e[e.length - 1];
    if (this.za)
      break;
    if (!n || n.node.type === "Program" && n.done) {
      if (!this.ca.length)
        return !1;
      if (n = this.ca[0], !n || n.time > Date.now())
        n = null;
      else {
        this.ca.shift(), 0 <= n.interval && Zr(this, n, n.interval);
        var r = new m(n.node, n.scope);
        n.zb && (r.ia = 2, r.B = this.Na, r.U = n.zb, r.Qa = !0, r.F = n.Lb), n = r;
      }
      if (!n)
        break;
    }
    r = n.node;
    var o = Ve;
    Ve = this;
    try {
      var s = this.rb[r.type](e, n, r);
    } catch (a) {
      if (a !== Kr)
        throw this.value !== a && (this.value = void 0), a;
    } finally {
      Ve = o;
    }
    if (s && e.push(s), this.P)
      throw this.value = void 0, Error("Getter not supported in this context");
    if (this.Ka)
      throw this.value = void 0, Error("Setter not supported in this context");
    t || r.end || (t = Date.now() + this.POLYFILL_TIMEOUT);
  } while (!r.end && t > Date.now());
  return !0;
};
b.Eb = function() {
  for (; !this.za && this.lb(); )
    ;
  return this.za;
};
function _n(e, t) {
  e.g(t, "NaN", NaN, Le), e.g(t, "Infinity", 1 / 0, Le), e.g(t, "undefined", void 0, Le), e.g(t, "window", t, Ht), e.g(t, "this", t, Le), e.g(t, "self", t), e.M = new j(null), e.Z = new j(e.M), Un(e, t), Ln(e, t), t.xa = e.M, e.g(t, "constructor", e.v, y), Fn(e, t), Wn(e, t), Gn(e, t), $n(e, t), Vn(e, t), Xn(e, t), Hn(e, t), Jn(e, t), Kn(e, t);
  var n = e.i(function() {
    throw EvalError("Can't happen");
  }, !1);
  n.eval = !0, e.g(t, "eval", n, y), e.g(t, "parseInt", e.i(parseInt, !1), y), e.g(t, "parseFloat", e.i(parseFloat, !1), y), e.g(t, "isNaN", e.i(isNaN, !1), y), e.g(t, "isFinite", e.i(isFinite, !1), y);
  for (var r = [[escape, "escape"], [unescape, "unescape"], [decodeURI, "decodeURI"], [decodeURIComponent, "decodeURIComponent"], [encodeURI, "encodeURI"], [encodeURIComponent, "encodeURIComponent"]], o = 0; o < r.length; o++)
    n = function(s) {
      return function(a) {
        try {
          return s(a);
        } catch (f) {
          w(e, e.Gb, f.message);
        }
      };
    }(r[o][0]), e.g(t, r[o][1], e.i(n, !1), y);
  n = function(s) {
    return Lr(e, !1, arguments);
  }, e.g(t, "setTimeout", e.i(n, !1), y), n = function(s) {
    return Lr(e, !0, arguments);
  }, e.g(t, "setInterval", e.i(n, !1), y), n = function(s) {
    Fr(e, s);
  }, e.g(t, "clearTimeout", e.i(n, !1), y), n = function(s) {
    Fr(e, s);
  }, e.g(t, "clearInterval", e.i(n, !1), y), e.OBJECT = e.v, e.OBJECT_PROTO = e.M, e.FUNCTION = e.O, e.FUNCTION_PROTO = e.Z, e.ARRAY = e.qa, e.ARRAY_PROTO = e.La, e.REGEXP = e.I, e.REGEXP_PROTO = e.Ma, e.DATE = e.$, e.DATE_PROTO = e.nb, e.qb && e.qb(e, t);
}
b.Wb = 0;
function Un(e, t) {
  var n = /^[A-Za-z_$][\w$]*$/, r = function(o) {
    var s = arguments.length ? String(arguments[arguments.length - 1]) : "", a = Array.prototype.slice.call(arguments, 0, -1).join(",").trim();
    if (a) {
      a = a.split(/\s*,\s*/);
      for (var f = 0; f < a.length; f++) {
        var d = a[f];
        n.test(d) || w(e, e.T, "Invalid function argument: " + d);
      }
      a = a.join(", ");
    }
    try {
      var v = Oe("(function(" + a + ") {" + s + "})", "function" + e.Wb++);
    } catch (I) {
      w(e, e.T, "Invalid code: " + I.message);
    }
    return v.body.length !== 1 && w(e, e.T, "Invalid code in function body"), Yt(e, v.body[0].la, e.N, "anonymous");
  };
  e.O = e.i(r, !0), e.g(t, "Function", e.O, y), e.g(e.O, "prototype", e.Z, y), e.g(e.Z, "constructor", e.O, y), e.Z.Ta = function() {
  }, e.Z.Ta.id = e.Ya++, e.Z.Ab = !0, e.g(e.Z, "length", 0, K), e.Z.H = "Function", r = function(o, s) {
    var a = e.o[e.o.length - 1];
    a.U = this, a.B = o, a.F = [], s != null && (s instanceof j ? a.F = zr(e, s) : w(e, e.j, "CreateListFromArrayLike called on non-object")), a.eb = !1;
  }, F(e, e.O, "apply", r), r = function(o) {
    var s = e.o[e.o.length - 1];
    s.U = this, s.B = o, s.F = [];
    for (var a = 1; a < arguments.length; a++)
      s.F.push(arguments[a]);
    s.eb = !1;
  }, F(e, e.O, "call", r), e.ba.push("Object.defineProperty(Function.prototype, 'bind',", "{configurable: true, writable: true, value:", "function bind(oThis) {", "if (typeof this !== 'function') {", "throw TypeError('What is trying to be bound is not callable');", "}", "var aArgs   = Array.prototype.slice.call(arguments, 1),", "fToBind = this,", "fNOP    = function() {},", "fBound  = function() {", "return fToBind.apply(this instanceof fNOP", "? this", ": oThis,", "aArgs.concat(Array.prototype.slice.call(arguments)));", "};", "if (this.prototype) {", "fNOP.prototype = this.prototype;", "}", "fBound.prototype = new fNOP();", "return fBound;", "}", "});", ""), r = function() {
    return String(this);
  }, F(e, e.O, "toString", r), e.g(e.O, "toString", e.i(r, !1), y), r = function() {
    return this.valueOf();
  }, F(e, e.O, "valueOf", r), e.g(e.O, "valueOf", e.i(r, !1), y);
}
function Ln(e, t) {
  function n(o) {
    o != null || w(e, e.j, "Cannot convert '" + o + "' to object");
  }
  var r = function(o) {
    if (o == null)
      return le(e) ? this : e.s(e.M);
    if (!(o instanceof j)) {
      var s = e.s(ae(e, o));
      return s.data = o, s;
    }
    return o;
  };
  e.v = e.i(r, !0), e.g(e.v, "prototype", e.M, y), e.g(e.M, "constructor", e.v, y), e.g(t, "Object", e.v, y), r = function(o) {
    return n(o), ie(e, Object.getOwnPropertyNames(o instanceof j ? o.h : o));
  }, e.g(e.v, "getOwnPropertyNames", e.i(r, !1), y), r = function(o) {
    return n(o), o instanceof j && (o = o.h), ie(e, Object.keys(o));
  }, e.g(e.v, "keys", e.i(r, !1), y), r = function(o) {
    return o === null ? e.s(null) : (o instanceof j || w(e, e.j, "Object prototype may only be an Object or null, not " + o), e.s(o));
  }, e.g(e.v, "create", e.i(r, !1), y), e.ba.push("(function() {", "var create_ = Object.create;", "Object.create = function create(proto, props) {", "var obj = create_(proto);", "props && Object.defineProperties(obj, props);", "return obj;", "};", "})();", ""), r = function(o, s, a) {
    return s = String(s), o instanceof j || w(e, e.j, "Object.defineProperty called on non-object: " + o), a instanceof j || w(e, e.j, "Property description must be an object"), !o.preventExtensions || s in o.h || w(e, e.j, "Can't define property '" + s + "', object is not extensible"), e.g(o, s, Je, a.h), o;
  }, e.g(e.v, "defineProperty", e.i(r, !1), y), e.ba.push("(function() {", "var defineProperty_ = Object.defineProperty;", "Object.defineProperty = function defineProperty(obj, prop, d1) {", "var d2 = {};", "if ('configurable' in d1) d2.configurable = d1.configurable;", "if ('enumerable' in d1) d2.enumerable = d1.enumerable;", "if ('writable' in d1) d2.writable = d1.writable;", "if ('value' in d1) d2.value = d1.value;", "if ('get' in d1) d2.get = d1.get;", "if ('set' in d1) d2.set = d1.set;", "return defineProperty_(obj, prop, d2);", "};", "})();", "Object.defineProperty(Object, 'defineProperties',", "{configurable: true, writable: true, value:", "function defineProperties(obj, props) {", "var keys = Object.keys(props);", "for (var i = 0; i < keys.length; i++) {", "Object.defineProperty(obj, keys[i], props[keys[i]]);", "}", "return obj;", "}", "});", ""), r = function(o, s) {
    if (o instanceof j || w(e, e.j, "Object.getOwnPropertyDescriptor called on non-object: " + o), s = String(s), s in o.h) {
      var a = Object.getOwnPropertyDescriptor(o.h, s), f = o.V[s];
      return o = o.Y[s], s = e.s(e.M), f || o ? (e.g(s, "get", f), e.g(s, "set", o)) : (e.g(s, "value", a.value), e.g(s, "writable", a.writable)), e.g(s, "configurable", a.configurable), e.g(s, "enumerable", a.enumerable), s;
    }
  }, e.g(e.v, "getOwnPropertyDescriptor", e.i(r, !1), y), r = function(o) {
    return n(o), ae(e, o);
  }, e.g(e.v, "getPrototypeOf", e.i(r, !1), y), r = function(o) {
    return !!o && !o.preventExtensions;
  }, e.g(e.v, "isExtensible", e.i(r, !1), y), r = function(o) {
    return o instanceof j && (o.preventExtensions = !0), o;
  }, e.g(e.v, "preventExtensions", e.i(r, !1), y), F(e, e.v, "toString", j.prototype.toString), F(e, e.v, "toLocaleString", j.prototype.toString), F(e, e.v, "valueOf", j.prototype.valueOf), r = function(o) {
    return n(this), this instanceof j ? String(o) in this.h : this.hasOwnProperty(o);
  }, F(e, e.v, "hasOwnProperty", r), r = function(o) {
    return n(this), this instanceof j ? Object.prototype.propertyIsEnumerable.call(this.h, o) : this.propertyIsEnumerable(o);
  }, F(e, e.v, "propertyIsEnumerable", r), r = function(o) {
    for (; ; ) {
      if (o = ae(e, o), !o)
        return !1;
      if (o === this)
        return !0;
    }
  }, F(e, e.v, "isPrototypeOf", r);
}
function Fn(e, t) {
  var n = function(r) {
    var o = le(e) ? this : je(e), s = arguments[0];
    if (arguments.length === 1 && typeof s == "number")
      isNaN(Jt(s)) && w(e, e.ob, "Invalid array length: " + s), o.h.length = s;
    else {
      for (s = 0; s < arguments.length; s++)
        o.h[s] = arguments[s];
      o.h.length = s;
    }
    return o;
  };
  e.qa = e.i(n, !0), e.La = e.qa.h.prototype, e.g(t, "Array", e.qa, y), n = function(r) {
    return r && r.H === "Array";
  }, e.g(e.qa, "isArray", e.i(n, !1), y), e.g(e.La, "length", 0, {
    configurable: !1,
    enumerable: !1,
    writable: !0
  }), e.La.H = "Array", e.ba.push("(function() {", "function createArrayMethod_(f) {", "Object.defineProperty(Array.prototype, f.name,", "{configurable: true, writable: true, value: f});", "}", "createArrayMethod_(", "function pop() {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "if (!len || len < 0) {", "o.length = 0;", "return undefined;", "}", "len--;", "var x = o[len];", "delete o[len];", "o.length = len;", "return x;", "}", ");", "createArrayMethod_(", "function push(var_args) {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "for (var i = 0; i < arguments.length; i++) {", "o[len] = arguments[i];", "len++;", "}", "o.length = len;", "return len;", "}", ");", "createArrayMethod_(", "function shift() {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "if (!len || len < 0) {", "o.length = 0;", "return undefined;", "}", "var value = o[0];", "for (var i = 0; i < len - 1; i++) {", "if ((i + 1) in o) {", "o[i] = o[i + 1];", "} else {", "delete o[i];", "}", "}", "delete o[i];", "o.length = len - 1;", "return value;", "}", ");", "createArrayMethod_(", "function unshift(var_args) {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "if (!len || len < 0) {", "len = 0;", "}", "for (var i = len - 1; i >= 0; i--) {", "if (i in o) {", "o[i + arguments.length] = o[i];", "} else {", "delete o[i + arguments.length];", "}", "}", "for (var i = 0; i < arguments.length; i++) {", "o[i] = arguments[i];", "}", "return (o.length = len + arguments.length);", "}", ");", "createArrayMethod_(", "function reverse() {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "if (!len || len < 2) {", "return o;", "}", "for (var i = 0; i < len / 2 - 0.5; i++) {", "var x = o[i];", "var hasX = i in o;", "if ((len - i - 1) in o) {", "o[i] = o[len - i - 1];", "} else {", "delete o[i];", "}", "if (hasX) {", "o[len - i - 1] = x;", "} else {", "delete o[len - i - 1];", "}", "}", "return o;", "}", ");", "createArrayMethod_(", "function indexOf(searchElement, fromIndex) {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "var n = fromIndex | 0;", "if (!len || n >= len) {", "return -1;", "}", "var i = Math.max(n >= 0 ? n : len - Math.abs(n), 0);", "while (i < len) {", "if (i in o && o[i] === searchElement) {", "return i;", "}", "i++;", "}", "return -1;", "}", ");", "createArrayMethod_(", "function lastIndexOf(searchElement, fromIndex) {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "if (!len) {", "return -1;", "}", "var n = len - 1;", "if (arguments.length > 1) {", "n = fromIndex | 0;", "if (n) {", "n = (n > 0 || -1) * Math.floor(Math.abs(n));", "}", "}", "var i = n >= 0 ? Math.min(n, len - 1) : len - Math.abs(n);", "while (i >= 0) {", "if (i in o && o[i] === searchElement) {", "return i;", "}", "i--;", "}", "return -1;", "}", ");", "createArrayMethod_(", "function slice(start, end) {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "start |= 0;", "start = (start >= 0) ? start : Math.max(0, len + start);", "if (typeof end !== 'undefined') {", "if (end !== Infinity) {", "end |= 0;", "}", "if (end < 0) {", "end = len + end;", "} else {", "end = Math.min(end, len);", "}", "} else {", "end = len;", "}", "var size = end - start;", "var cloned = new Array(size);", "for (var i = 0; i < size; i++) {", "if ((start + i) in o) {", "cloned[i] = o[start + i];", "}", "}", "return cloned;", "}", ");", "createArrayMethod_(", "function splice(start, deleteCount, var_args) {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "start |= 0;", "if (start < 0) {", "start = Math.max(len + start, 0);", "} else {", "start = Math.min(start, len);", "}", "if (arguments.length < 1) {", "deleteCount = len - start;", "} else {", "deleteCount |= 0;", "deleteCount = Math.max(0, Math.min(deleteCount, len - start));", "}", "var removed = [];", "for (var i = start; i < start + deleteCount; i++) {", "if (i in o) {", "removed.push(o[i]);", "} else {", "removed.length++;", "}", "if ((i + deleteCount) in o) {", "o[i] = o[i + deleteCount];", "} else {", "delete o[i];", "}", "}", "for (var i = start + deleteCount; i < len - deleteCount; i++) {", "if ((i + deleteCount) in o) {", "o[i] = o[i + deleteCount];", "} else {", "delete o[i];", "}", "}", "for (var i = len - deleteCount; i < len; i++) {", "delete o[i];", "}", "len -= deleteCount;", "var arl = arguments.length - 2;", "for (var i = len - 1; i >= start; i--) {", "if (i in o) {", "o[i + arl] = o[i];", "} else {", "delete o[i + arl];", "}", "}", "len += arl;", "for (var i = 2; i < arguments.length; i++) {", "o[start + i - 2] = arguments[i];", "}", "o.length = len;", "return removed;", "}", ");", "createArrayMethod_(", "function concat(var_args) {", "if (!this) throw TypeError();", "var o = Object(this);", "var cloned = [];", "for (var i = -1; i < arguments.length; i++) {", "var value = (i === -1) ? o : arguments[i];", "if (Array.isArray(value)) {", "for (var j = 0, l = value.length; j < l; j++) {", "if (j in value) {", "cloned.push(value[j]);", "} else {", "cloned.length++;", "}", "}", "} else {", "cloned.push(value);", "}", "}", "return cloned;", "}", ");", "createArrayMethod_(", "function join(opt_separator) {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "var sep = typeof opt_separator === 'undefined' ?", "',' : ('' + opt_separator);", "var str = '';", "for (var i = 0; i < len; i++) {", "if (i && sep) str += sep;", "str += (o[i] === null || o[i] === undefined) ? '' : o[i];", "}", "return str;", "}", ");", "createArrayMethod_(", "function every(callbackfn, thisArg) {", "if (!this || typeof callbackfn !== 'function') throw TypeError();", "var t, k = 0;", "var o = Object(this), len = o.length >>> 0;", "if (arguments.length > 1) t = thisArg;", "while (k < len) {", "if (k in o && !callbackfn.call(t, o[k], k, o)) return false;", "k++;", "}", "return true;", "}", ");", "createArrayMethod_(", "function filter(fun, var_args) {", "if (this === void 0 || this === null || typeof fun !== 'function') throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "var res = [];", "var thisArg = arguments.length >= 2 ? arguments[1] : void 0;", "for (var i = 0; i < len; i++) {", "if (i in o) {", "var val = o[i];", "if (fun.call(thisArg, val, i, o)) res.push(val);", "}", "}", "return res;", "}", ");", "createArrayMethod_(", "function forEach(callback, thisArg) {", "if (!this || typeof callback !== 'function') throw TypeError();", "var t, k = 0;", "var o = Object(this), len = o.length >>> 0;", "if (arguments.length > 1) t = thisArg;", "while (k < len) {", "if (k in o) callback.call(t, o[k], k, o);", "k++;", "}", "}", ");", "createArrayMethod_(", "function map(callback, thisArg) {", "if (!this || typeof callback !== 'function') throw TypeError();", "var t, k = 0;", "var o = Object(this), len = o.length >>> 0;", "if (arguments.length > 1) t = thisArg;", "var a = new Array(len);", "while (k < len) {", "if (k in o) a[k] = callback.call(t, o[k], k, o);", "k++;", "}", "return a;", "}", ");", "createArrayMethod_(", "function reduce(callback /*, initialValue*/) {", "if (!this || typeof callback !== 'function') throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "var k = 0, value;", "if (arguments.length === 2) {", "value = arguments[1];", "} else {", "while (k < len && !(k in o)) k++;", "if (k >= len) {", "throw TypeError('Reduce of empty array with no initial value');", "}", "value = o[k++];", "}", "for (; k < len; k++) {", "if (k in o) value = callback(value, o[k], k, o);", "}", "return value;", "}", ");", "createArrayMethod_(", "function reduceRight(callback /*, initialValue*/) {", "if (null === this || 'undefined' === typeof this || 'function' !== typeof callback) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "var k = len - 1, value;", "if (arguments.length >= 2) {", "value = arguments[1];", "} else {", "while (k >= 0 && !(k in o)) k--;", "if (k < 0) {", "throw TypeError('Reduce of empty array with no initial value');", "}", "value = o[k--];", "}", "for (; k >= 0; k--) {", "if (k in o) value = callback(value, o[k], k, o);", "}", "return value;", "}", ");", "createArrayMethod_(", "function some(fun/*, thisArg*/) {", "if (!this || typeof fun !== 'function') throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "var thisArg = arguments.length >= 2 ? arguments[1] : void 0;", "for (var i = 0; i < len; i++) {", "if (i in o && fun.call(thisArg, o[i], i, o)) return true;", "}", "return false;", "}", ");", "createArrayMethod_(", "function sort(opt_comp) {", "if (!this) throw TypeError();", "if (typeof opt_comp !== 'function') {", "opt_comp = undefined;", "}", "for (var i = 0; i < this.length; i++) {", "var changes = 0;", "for (var j = 0; j < this.length - i - 1; j++) {", "if (opt_comp ? (opt_comp(this[j], this[j + 1]) > 0) :", "(String(this[j]) > String(this[j + 1]))) {", "var swap = this[j];", "var hasSwap = j in this;", "if ((j + 1) in this) {", "this[j] = this[j + 1];", "} else {", "delete this[j];", "}", "if (hasSwap) {", "this[j + 1] = swap;", "} else {", "delete this[j + 1];", "}", "changes++;", "}", "}", "if (!changes) break;", "}", "return this;", "}", ");", "createArrayMethod_(", "function toLocaleString() {", "if (!this) throw TypeError();", "var o = Object(this), len = o.length >>> 0;", "var out = [];", "for (var i = 0; i < len; i++) {", "out[i] = (o[i] === null || o[i] === undefined) ? '' : o[i].toLocaleString();", "}", "return out.join(',');", "}", ");", "})();", "");
}
function Wn(e, t) {
  var n = function(r) {
    return r = arguments.length ? pe.String(r) : "", le(e) ? (this.data = r, this) : r;
  };
  for (e.J = e.i(n, !0), e.g(t, "String", e.J, y), e.g(e.J, "fromCharCode", e.i(String.fromCharCode, !1), y), t = "charAt charCodeAt concat indexOf lastIndexOf slice substr substring toLocaleLowerCase toLocaleUpperCase toLowerCase toUpperCase trim".split(" "), n = 0; n < t.length; n++)
    F(e, e.J, t[n], String.prototype[t[n]]);
  n = function(r, o, s) {
    o = e.R(o), s = e.R(s);
    try {
      return String(this).localeCompare(r, o, s);
    } catch (a) {
      w(e, e.D, "localeCompare: " + a.message);
    }
  }, F(e, e.J, "localeCompare", n), n = function(r, o, s) {
    var a = String(this);
    if (o = o ? Number(o) : void 0, G(e, r, e.I) && (r = r.data, We(e, r, s), e.REGEXP_MODE === 2)) {
      if (se)
        r = Fe(e, "string.split(separator, limit)", {
          string: a,
          separator: r,
          limit: o
        }, r, s), r !== ke && s(ie(e, r));
      else {
        var f = e.ha(), d = Ge(e, r, f, s);
        f.onmessage = function(v) {
          clearTimeout(d), s(ie(e, v.data));
        }, f.postMessage(["split", a, r, o]);
      }
      return;
    }
    r = a.split(r, o), s(ie(e, r));
  }, $e(e, e.J, "split", n), n = function(r, o) {
    var s = String(this);
    if (r = G(e, r, e.I) ? r.data : new RegExp(r), We(e, r, o), e.REGEXP_MODE === 2)
      if (se)
        r = Fe(e, "string.match(regexp)", {
          string: s,
          regexp: r
        }, r, o), r !== ke && o(r && ie(e, r));
      else {
        var a = e.ha(), f = Ge(e, r, a, o);
        a.onmessage = function(d) {
          clearTimeout(f), o(d.data && ie(e, d.data));
        }, a.postMessage(["match", s, r]);
      }
    else
      r = s.match(r), o(r && ie(e, r));
  }, $e(e, e.J, "match", n), n = function(r, o) {
    var s = String(this);
    if (G(e, r, e.I) ? r = r.data : r = new RegExp(r), We(e, r, o), e.REGEXP_MODE === 2)
      if (se)
        r = Fe(e, "string.search(regexp)", {
          string: s,
          regexp: r
        }, r, o), r !== ke && o(r);
      else {
        var a = e.ha(), f = Ge(e, r, a, o);
        a.onmessage = function(d) {
          clearTimeout(f), o(d.data);
        }, a.postMessage(["search", s, r]);
      }
    else
      o(s.search(r));
  }, $e(e, e.J, "search", n), n = function(r, o, s) {
    var a = String(this);
    if (o = String(o), G(e, r, e.I) && (r = r.data, We(e, r, s), e.REGEXP_MODE === 2)) {
      if (se)
        r = Fe(e, "string.replace(substr, newSubstr)", {
          string: a,
          substr: r,
          newSubstr: o
        }, r, s), r !== ke && s(r);
      else {
        var f = e.ha(), d = Ge(e, r, f, s);
        f.onmessage = function(v) {
          clearTimeout(d), s(v.data);
        }, f.postMessage(["replace", a, r, o]);
      }
      return;
    }
    s(a.replace(r, o));
  }, $e(e, e.J, "replace", n), e.ba.push("(function() {", "var replace_ = String.prototype.replace;", "String.prototype.replace = function replace(substr, newSubstr) {", "if (typeof newSubstr !== 'function') {", "return replace_.call(this, substr, newSubstr);", "}", "var str = this;", "if (substr instanceof RegExp) {", "var subs = [];", "var m = substr.exec(str);", "while (m) {", "m.push(m.index, str);", "var inject = newSubstr.apply(null, m);", "subs.push([m.index, m[0].length, inject]);", "m = substr.global ? substr.exec(str) : null;", "}", "for (var i = subs.length - 1; i >= 0; i--) {", "str = str.substring(0, subs[i][0]) + subs[i][2] + str.substring(subs[i][0] + subs[i][1]);", "}", "} else {", "var i = str.indexOf(substr);", "if (i !== -1) {", "var inject = newSubstr(str.substr(i, substr.length), i, str);", "str = str.substring(0, i) + inject + str.substring(i + substr.length);", "}", "}", "return str;", "};", "})();", "");
}
function Gn(e, t) {
  e.mb = e.i(function(n) {
    return n = pe.Boolean(n), le(e) ? (this.data = n, this) : n;
  }, !0), e.g(t, "Boolean", e.mb, y);
}
function $n(e, t) {
  var n = function(r) {
    return r = arguments.length ? pe.Number(r) : 0, le(e) ? (this.data = r, this) : r;
  };
  for (e.aa = e.i(n, !0), e.g(t, "Number", e.aa, y), t = ["MAX_VALUE", "MIN_VALUE", "NaN", "NEGATIVE_INFINITY", "POSITIVE_INFINITY"], n = 0; n < t.length; n++)
    e.g(e.aa, t[n], Number[t[n]], Le);
  n = function(r) {
    try {
      return Number(this).toExponential(r);
    } catch (o) {
      w(e, e.D, o.message);
    }
  }, F(e, e.aa, "toExponential", n), n = function(r) {
    try {
      return Number(this).toFixed(r);
    } catch (o) {
      w(e, e.D, o.message);
    }
  }, F(e, e.aa, "toFixed", n), n = function(r) {
    try {
      return Number(this).toPrecision(r);
    } catch (o) {
      w(e, e.D, o.message);
    }
  }, F(e, e.aa, "toPrecision", n), n = function(r) {
    try {
      return Number(this).toString(r);
    } catch (o) {
      w(e, e.D, o.message);
    }
  }, F(e, e.aa, "toString", n), n = function(r, o) {
    r = r ? e.R(r) : void 0, o = o ? e.R(o) : void 0;
    try {
      return Number(this).toLocaleString(r, o);
    } catch (s) {
      w(e, e.D, "toLocaleString: " + s.message);
    }
  }, F(e, e.aa, "toLocaleString", n);
}
function Vn(e, t) {
  var n = function(o, s) {
    if (!le(e))
      return pe.Date();
    var a = [null].concat(Array.from(arguments));
    return this.data = new (Function.prototype.bind.apply(pe.Date, a))(), this;
  };
  e.$ = e.i(n, !0), e.nb = e.$.h.prototype, e.g(t, "Date", e.$, y), e.g(e.$, "now", e.i(Date.now, !1), y), e.g(e.$, "parse", e.i(Date.parse, !1), y), e.g(e.$, "UTC", e.i(Date.UTC, !1), y), t = "getDate getDay getFullYear getHours getMilliseconds getMinutes getMonth getSeconds getTime getTimezoneOffset getUTCDate getUTCDay getUTCFullYear getUTCHours getUTCMilliseconds getUTCMinutes getUTCMonth getUTCSeconds getYear setDate setFullYear setHours setMilliseconds setMinutes setMonth setSeconds setTime setUTCDate setUTCFullYear setUTCHours setUTCMilliseconds setUTCMinutes setUTCMonth setUTCSeconds setYear toDateString toISOString toJSON toGMTString toLocaleDateString toLocaleString toLocaleTimeString toTimeString toUTCString".split(" ");
  for (var r = 0; r < t.length; r++)
    n = function(o) {
      return function(s) {
        var a = this.data;
        a instanceof Date || w(e, e.j, o + " not called on a Date");
        for (var f = [], d = 0; d < arguments.length; d++)
          f[d] = e.R(arguments[d]);
        return a[o].apply(a, f);
      };
    }(t[r]), F(e, e.$, t[r], n);
}
function Xn(e, t) {
  var n = function(r, o) {
    if (le(e))
      var s = this;
    else {
      if (o === void 0 && G(e, r, e.I))
        return r;
      s = e.s(e.Ma);
    }
    r = r === void 0 ? "" : String(r), o = o ? String(o) : "", /^[gmi]*$/.test(o) || w(e, e.T, "Invalid regexp flag: " + o);
    try {
      var a = new pe.RegExp(r, o);
    } catch (f) {
      w(e, e.T, f.message);
    }
    return Kt(e, s, a), s;
  };
  e.I = e.i(n, !0), e.Ma = e.I.h.prototype, e.g(t, "RegExp", e.I, y), e.g(e.I.h.prototype, "global", void 0, K), e.g(e.I.h.prototype, "ignoreCase", void 0, K), e.g(e.I.h.prototype, "multiline", void 0, K), e.g(e.I.h.prototype, "source", "(?:)", K), e.ba.push("Object.defineProperty(RegExp.prototype, 'test',", "{configurable: true, writable: true, value:", "function test(str) {", "return !!this.exec(str);", "}", "});"), n = function(r, o) {
    function s(I) {
      if (I) {
        var A = ie(e, I);
        return e.g(A, "index", I.index), e.g(A, "input", I.input), A;
      }
      return null;
    }
    var a = this.data;
    if (r = String(r), a.lastIndex = Number(e.G(this, "lastIndex")), We(e, a, o), e.REGEXP_MODE === 2)
      if (se)
        r = Fe(e, "regexp.exec(string)", {
          string: r,
          regexp: a
        }, a, o), r !== ke && (e.g(this, "lastIndex", a.lastIndex), o(s(r)));
      else {
        var f = e.ha(), d = Ge(e, a, f, o), v = this;
        f.onmessage = function(I) {
          clearTimeout(d), e.g(v, "lastIndex", I.data[1]), o(s(I.data[0]));
        }, f.postMessage(["exec", a, a.lastIndex, r]);
      }
    else
      r = a.exec(r), e.g(this, "lastIndex", a.lastIndex), o(s(r));
  }, $e(e, e.I, "exec", n);
}
function Hn(e, t) {
  function n(r) {
    var o = e.i(function(s) {
      var a = le(e) ? this : e.Aa(o);
      return Gt(e, a, s), a;
    }, !0);
    return e.g(o, "prototype", e.Aa(e.D), y), e.g(o.h.prototype, "name", r, y), e.g(t, r, o, y), o;
  }
  e.D = e.i(function(r) {
    var o = le(e) ? this : e.Aa(e.D);
    return Gt(e, o, r), o;
  }, !0), e.g(t, "Error", e.D, y), e.g(e.D.h.prototype, "message", "", y), e.g(e.D.h.prototype, "name", "Error", y), n("EvalError"), e.ob = n("RangeError"), e.pb = n("ReferenceError"), e.T = n("SyntaxError"), e.j = n("TypeError"), e.Gb = n("URIError");
}
function Jn(e, t) {
  var n = e.s(e.M);
  e.g(t, "Math", n, y);
  var r = "E LN2 LN10 LOG2E LOG10E PI SQRT1_2 SQRT2".split(" ");
  for (t = 0; t < r.length; t++)
    e.g(n, r[t], Math[r[t]], K);
  for (r = "abs acos asin atan atan2 ceil cos exp floor log max min pow random round sin sqrt tan".split(" "), t = 0; t < r.length; t++)
    e.g(n, r[t], e.i(Math[r[t]], !1), y);
}
function Kn(e, t) {
  var n = e.s(e.M);
  e.g(t, "JSON", n, y), t = function(r) {
    try {
      var o = JSON.parse(String(r));
    } catch (s) {
      w(e, e.T, s.message);
    }
    return e.Ia(o);
  }, e.g(n, "parse", e.i(t, !1)), t = function(r, o, s) {
    o && o.H === "Function" ? w(e, e.j, "Function replacer on JSON.stringify not supported") : o && o.H === "Array" ? (o = zr(e, o), o = o.filter(function(f) {
      return typeof f == "string" || typeof f == "number";
    })) : o = null, typeof s != "string" && typeof s != "number" && (s = void 0), r = e.R(r);
    try {
      var a = JSON.stringify(r, o, s);
    } catch (f) {
      w(e, e.j, f.message);
    }
    return a;
  }, e.g(n, "stringify", e.i(t, !1));
}
function G(e, t, n) {
  if (t == null || !n)
    return !1;
  if (n = n.h.prototype, t === n)
    return !0;
  for (t = ae(e, t); t; ) {
    if (t === n)
      return !0;
    t = t.xa;
  }
  return !1;
}
function Kt(e, t, n) {
  t.data = new RegExp(n.source, n.flags), e.g(t, "lastIndex", n.lastIndex, y), e.g(t, "source", n.source, K), e.g(t, "global", n.global, K), e.g(t, "ignoreCase", n.ignoreCase, K), e.g(t, "multiline", n.multiline, K);
}
function Gt(e, t, n) {
  n && e.g(t, "message", String(n), y), n = [];
  for (var r = e.o.length - 1; 0 <= r; r--) {
    var o = e.o[r], s = o.node;
    s.type === "CallExpression" && (o = o.U) && n.length && (n[n.length - 1].Ob = e.G(o, "name")), !s.X || n.length && s.type !== "CallExpression" || n.push({
      Nb: s.X
    });
  }
  for (r = String(e.G(t, "name")), s = String(e.G(t, "message")), s = r + ": " + s + `
`, r = 0; r < n.length; r++) {
    var a = n[r].Nb;
    o = n[r].Ob, a = a.source + ":" + a.start.line + ":" + a.start.ab, s = o ? s + ("  at " + o + " (" + a + `)
`) : s + ("  at " + a + `
`);
  }
  e.g(t, "stack", s.trim(), y);
}
b.ha = function() {
  var e = this.ha.Mb;
  return e || (e = new Blob([Dn.join(`
`)], {
    type: "application/javascript"
  }), this.ha.Mb = e), new Worker(URL.createObjectURL(e));
};
function Fe(e, t, n, r, o) {
  var s = {
    timeout: e.REGEXP_THREAD_TIMEOUT
  };
  try {
    return se.runInNewContext(t, n, s);
  } catch {
    o(null), w(e, e.D, "RegExp Timeout: " + r);
  }
  return ke;
}
function We(e, t, n) {
  if (e.REGEXP_MODE === 0)
    var r = !1;
  else if (e.REGEXP_MODE === 1)
    r = !0;
  else if (se)
    r = !0;
  else if (typeof Worker == "function" && typeof URL == "function")
    r = !0;
  else if (typeof require == "function") {
    try {
      se = require("vm");
    } catch {
    }
    r = !!se;
  } else
    r = !1;
  r || (n(null), w(e, e.D, "Regular expressions not supported: " + t));
}
function Ge(e, t, n, r) {
  return setTimeout(function() {
    n.terminate(), r(null);
    try {
      w(e, e.D, "RegExp Timeout: " + t);
    } catch {
    }
  }, e.REGEXP_THREAD_TIMEOUT);
}
b.Aa = function(e) {
  return this.s(e && e.h.prototype);
};
b.s = function(e) {
  if (typeof e != "object")
    throw Error("Non object prototype");
  return e = new j(e), G(this, e, this.D) && (e.H = "Error"), e;
};
function je(e) {
  var t = e.s(e.La);
  return e.g(t, "length", 0, {
    configurable: !1,
    enumerable: !1,
    writable: !0
  }), t.H = "Array", t;
}
function zt(e, t, n) {
  var r = e.s(e.Z);
  return n ? (n = e.s(e.M), e.g(r, "prototype", n, y), e.g(n, "constructor", r, y)) : r.Ab = !0, e.g(r, "length", t, K), r.H = "Function", r;
}
function Yt(e, t, n, r) {
  var o = zt(e, t.oa.length, !0);
  return o.Va = n, o.node = t, e.g(o, "name", t.id ? String(t.id.name) : r || "", K), o;
}
b.i = function(e, t) {
  return t = zt(this, e.length, t), t.Ta = e, e.id = this.Ya++, this.g(t, "name", e.name, K), t;
};
b.ub = function(e) {
  var t = zt(this, e.length, !0);
  return t.Za = e, e.id = this.Ya++, this.g(t, "name", e.name, K), t;
};
b.Ia = function(e) {
  if (e instanceof j)
    throw Error("Object is already pseudo");
  if (e == null || e === !0 || e === !1 || typeof e == "string" || typeof e == "number")
    return e;
  if (e instanceof RegExp) {
    var t = this.s(this.Ma);
    return Kt(this, t, e), t;
  }
  if (e instanceof Date)
    return t = this.s(this.nb), t.data = new Date(e.valueOf()), t;
  if (typeof e == "function") {
    var n = this;
    return t = Object.getOwnPropertyDescriptor(e, "prototype"), this.i(function() {
      var o = Array.prototype.slice.call(arguments).map(function(s) {
        return n.R(s);
      });
      return o = e.apply(n, o), n.Ia(o);
    }, !!t);
  }
  if (Array.isArray(e)) {
    t = je(this);
    for (var r = 0; r < e.length; r++)
      r in e && this.g(t, r, this.Ia(e[r]));
    return t;
  }
  t = this.s(this.M);
  for (r in e)
    this.g(t, r, this.Ia(e[r]));
  return t;
};
b.R = function(e, t) {
  if (typeof e != "object" && typeof e != "function" || e === null)
    return e;
  if (!(e instanceof j))
    throw Error("Object is not pseudo");
  if (G(this, e, this.I))
    return t = new RegExp(e.data.source, e.data.flags), t.lastIndex = e.data.lastIndex, t;
  if (G(this, e, this.$))
    return new Date(e.data.valueOf());
  t = t || {
    hb: [],
    Sa: []
  };
  var n = t.hb.indexOf(e);
  if (n !== -1)
    return t.Sa[n];
  if (t.hb.push(e), G(this, e, this.qa)) {
    n = [], t.Sa.push(n);
    for (var r = this.G(e, "length"), o = 0; o < r; o++)
      ct(this, e, o) && (n[o] = this.R(this.G(e, o), t));
  } else
    for (r in n = {}, t.Sa.push(n), e.h)
      o = this.R(e.h[r], t), Object.defineProperty(n, r, {
        value: o,
        writable: !0,
        enumerable: !0,
        configurable: !0
      });
  return t.hb.pop(), t.Sa.pop(), n;
};
function ie(e, t) {
  for (var n = je(e), r = Object.getOwnPropertyNames(t), o = 0; o < r.length; o++)
    e.g(n, r[o], t[r[o]]);
  return n;
}
function zr(e, t) {
  var n = [], r;
  for (r in t.h)
    n[r] = e.G(t, r);
  return n.length = Jt(e.G(t, "length")) || 0, n;
}
function ae(e, t) {
  switch (typeof t) {
    case "number":
      return e.aa.h.prototype;
    case "boolean":
      return e.mb.h.prototype;
    case "string":
      return e.J.h.prototype;
  }
  return t ? t.xa : null;
}
b.G = function(e, t) {
  if (this.P)
    throw Error("Getter not supported in that context");
  if (t = String(t), e != null || w(this, this.j, "Cannot read property '" + t + "' of " + e), typeof e == "object" && !(e instanceof j))
    throw TypeError("Expecting native value or pseudo object");
  if (t === "length") {
    if (G(this, e, this.J))
      return String(e).length;
  } else if (64 > t.charCodeAt(0) && G(this, e, this.J)) {
    var n = Xe(t);
    if (!isNaN(n) && n < String(e).length)
      return String(e)[n];
  }
  do
    if (e.h && t in e.h)
      return (n = e.V[t]) ? (this.P = !0, n) : e.h[t];
  while (e = ae(this, e));
};
function ct(e, t, n) {
  if (!(t instanceof j))
    throw TypeError("Primitive data type has no properties");
  if (n = String(n), n === "length" && G(e, t, e.J))
    return !0;
  if (G(e, t, e.J)) {
    var r = Xe(n);
    if (!isNaN(r) && r < String(t).length)
      return !0;
  }
  do
    if (t.h && n in t.h)
      return !0;
  while (t = ae(e, t));
  return !1;
}
b.g = function(e, t, n, r) {
  if (this.Ka)
    throw Error("Setter not supported in that context");
  if (t = String(t), e != null || w(this, this.j, "Cannot set property '" + t + "' of " + e), typeof e == "object" && !(e instanceof j))
    throw TypeError("Expecting native value or pseudo object");
  r && ("get" in r || "set" in r) && ("value" in r || "writable" in r) && w(this, this.j, "Invalid property descriptor. Cannot both specify accessors and a value or writable attribute");
  var o = !this.o || Zt(this).S;
  if (e instanceof j) {
    if (G(this, e, this.J)) {
      var s = Xe(t);
      if (t === "length" || !isNaN(s) && s < String(e).length) {
        o && w(this, this.j, "Cannot assign to read only property '" + t + "' of String '" + e.data + "'");
        return;
      }
    }
    if (e.H === "Array")
      if (s = e.h.length, t === "length") {
        if (r) {
          if (!("value" in r))
            return;
          n = r.value;
        }
        if (n = Jt(n), isNaN(n) && w(this, this.ob, "Invalid array length"), n < s)
          for (a in e.h) {
            var a = Xe(a);
            !isNaN(a) && n <= a && delete e.h[a];
          }
      } else
        isNaN(a = Xe(t)) || (e.h.length = Math.max(s, a + 1));
    if (!e.preventExtensions || t in e.h)
      if (r) {
        o = {}, "get" in r && r.get && (e.V[t] = r.get, o.get = this.g.ac), "set" in r && r.set && (e.Y[t] = r.set, o.set = this.g.bc), "configurable" in r && (o.configurable = r.configurable), "enumerable" in r && (o.enumerable = r.enumerable), "writable" in r && (o.writable = r.writable, delete e.V[t], delete e.Y[t]), "value" in r ? (o.value = r.value, delete e.V[t], delete e.Y[t]) : n !== Je && (o.value = n, delete e.V[t], delete e.Y[t]);
        try {
          Object.defineProperty(e.h, t, o);
        } catch {
          w(this, this.j, "Cannot redefine property: " + t);
        }
        "get" in r && !r.get && delete e.V[t], "set" in r && !r.set && delete e.Y[t];
      } else {
        if (n === Je)
          throw ReferenceError("Value not specified");
        for (r = e; !(t in r.h); )
          if (r = ae(this, r), !r) {
            r = e;
            break;
          }
        if (r.Y && r.Y[t])
          return this.Ka = !0, r.Y[t];
        if (r.V && r.V[t])
          o && w(this, this.j, "Cannot set property '" + t + "' of object '" + e + "' which only has a getter");
        else
          try {
            e.h[t] = n;
          } catch {
            o && w(this, this.j, "Cannot assign to read only property '" + t + "' of object '" + e + "'");
          }
      }
    else
      o && w(this, this.j, "Can't add property '" + t + "', object is not extensible");
  } else
    o && w(this, this.j, "Can't create property '" + t + "' on '" + e + "'");
};
b.g.ac = function() {
  throw Error("Placeholder getter");
};
b.g.bc = function() {
  throw Error("Placeholder setter");
};
function F(e, t, n, r) {
  e.g(t.h.prototype, n, e.i(r, !1), y);
}
function $e(e, t, n, r) {
  e.g(t.h.prototype, n, e.ub(r), y);
}
function Zt(e) {
  if (e = e.o[e.o.length - 1].scope, !e)
    throw Error("No scope found");
  return e;
}
function $t(e, t, n) {
  var r = !1;
  if (n && n.S)
    r = !0;
  else {
    var o = t.body && t.body[0];
    o && o.la && o.la.type === "Literal" && o.la.value === "use strict" && (r = !0);
  }
  return o = e.s(null), r = new Qr(n, r, o), n || _n(e, r.object), Ke(e, t, r), r;
}
function Qt(e, t, n) {
  if (!t)
    throw Error("parentScope required");
  return e = n || e.s(null), new Qr(t, t.S, e);
}
function qt(e, t) {
  for (var n = Zt(e); n && n !== e.N; ) {
    if (t in n.object.h)
      return n.object.h[t];
    n = n.Va;
  }
  if (n === e.N && ct(e, n.object, t))
    return e.G(n.object, t);
  n = e.o[e.o.length - 1].node, n.type === "UnaryExpression" && n.operator === "typeof" || w(e, e.pb, t + " is not defined");
}
function Yr(e, t, n) {
  for (var r = Zt(e), o = r.S; r && r !== e.N; ) {
    if (t in r.object.h) {
      try {
        r.object.h[t] = n;
      } catch {
        o && w(e, e.j, "Cannot assign to read only variable '" + t + "'");
      }
      return;
    }
    r = r.Va;
  }
  if (r === e.N && (!o || ct(e, r.object, t)))
    return e.g(r.object, t, n);
  w(e, e.pb, t + " is not defined");
}
function Ke(e, t, n) {
  if (t.jb)
    var r = t.jb;
  else {
    switch (r = /* @__PURE__ */ Object.create(null), t.type) {
      case "VariableDeclaration":
        for (var o = 0; o < t.fa.length; o++)
          r[t.fa[o].id.name] = !0;
        break;
      case "FunctionDeclaration":
        r[t.id.name] = t;
        break;
      case "BlockStatement":
      case "CatchClause":
      case "DoWhileStatement":
      case "ForInStatement":
      case "ForStatement":
      case "IfStatement":
      case "LabeledStatement":
      case "Program":
      case "SwitchCase":
      case "SwitchStatement":
      case "TryStatement":
      case "WithStatement":
      case "WhileStatement":
        var s = t.constructor, a;
        for (a in t)
          if (a !== "loc") {
            var f = t[a];
            if (f && typeof f == "object") {
              if (Array.isArray(f)) {
                for (o = 0; o < f.length; o++)
                  if (f[o] && f[o].constructor === s) {
                    var d = Ke(e, f[o], n);
                    for (a in d)
                      r[a] = d[a];
                  }
              } else if (f.constructor === s)
                for (a in d = Ke(e, f, n), d)
                  r[a] = d[a];
            }
          }
    }
    t.jb = r;
  }
  for (a in r)
    r[a] === !0 ? e.g(n.object, a, void 0, _r) : e.g(n.object, a, Yt(e, r[a], n), _r);
  return r;
}
function le(e) {
  return e.o[e.o.length - 1].isConstructor;
}
function er(e, t) {
  return t[0] === Ae ? qt(e, t[1]) : e.G(t[0], t[1]);
}
function tr(e, t, n) {
  return t[0] === Ae ? Yr(e, t[1], n) : e.g(t[0], t[1], n);
}
function w(e, t, n) {
  throw e.N ? (n !== void 0 && t instanceof j && (t = e.Aa(t), Gt(e, t, n)), ze(e, 4, t), Kr) : n === void 0 ? t : n;
}
function ze(e, t, n, r) {
  if (t === 0)
    throw TypeError("Should not unwind for NORMAL completions");
  var o = e.o;
  e:
    for (; 0 < o.length; o.pop()) {
      var s = o[o.length - 1];
      switch (s.node.type) {
        case "TryStatement":
          s.ea = {
            type: t,
            value: n,
            label: r
          };
          return;
        case "CallExpression":
        case "NewExpression":
          if (t === 3) {
            s.value = n;
            return;
          }
          if (t !== 4)
            throw Error("Unsynatctic break/continue not rejected by Acorn");
          break;
        case "Program":
          s.done = !0;
          break e;
      }
      if (t === 1) {
        if (r ? s.labels && s.labels.indexOf(r) !== -1 : s.W || s.Zb) {
          o.pop();
          return;
        }
      } else if (t === 2 && (r ? s.labels && s.labels.indexOf(r) !== -1 : s.W))
        return;
    }
  throw G(e, n, e.D) ? (t = {
    EvalError,
    RangeError,
    ReferenceError,
    SyntaxError,
    TypeError,
    URIError
  }, r = String(e.G(n, "name")), o = e.G(n, "message").valueOf(), t = (t[r] || Error)(o), t.stack = String(e.G(n, "stack"))) : t = String(n), e.value = t, t;
}
function J(e, t) {
  switch (t.type) {
    case "ArrayExpression":
      return "[...]";
    case "BinaryExpression":
    case "LogicalExpression":
      return J(e, t.left) + " " + t.operator + " " + J(e, t.right);
    case "CallExpression":
      return J(e, t.callee) + "(...)";
    case "ConditionalExpression":
      return J(e, t.test) + " ? " + J(e, t.da) + " : " + J(e, t.alternate);
    case "Identifier":
      return t.name;
    case "Literal":
      return t.raw;
    case "MemberExpression":
      var n = J(e, t.object);
      return e = J(e, t.Wa), t.bb ? n + "[" + e + "]" : n + "." + e;
    case "NewExpression":
      return "new " + J(e, t.callee) + "(...)";
    case "ObjectExpression":
      return "{...}";
    case "ThisExpression":
      return "this";
    case "UnaryExpression":
      return t.operator + " " + J(e, t.K);
    case "UpdateExpression":
      return n = J(e, t.K), t.prefix ? t.operator + n : n + t.operator;
  }
  return "???";
}
function Lr(e, t, n) {
  var r = e.o[e.o.length - 1], o = Array.from(n), s = o.shift();
  n = Math.max(Number(o.shift() || 0), 0);
  var a = e.ya();
  if (s instanceof j && s.H === "Function") {
    var f = s;
    a.type = "CallExpression", r = r.scope;
  } else {
    try {
      var d = Oe(String(s), "taskCode" + e.ic++);
    } catch (v) {
      w(e, e.T, "Invalid code: " + v.message);
    }
    a.type = "EvalProgram_", a.body = d.body, r = r.node.arguments[0], ut(a, r ? r.start : void 0, r ? r.end : void 0), r = e.N, o.length = 0;
  }
  return t = new zn(f, o, r, a, t ? n : -1), Zr(e, t, n), t.Db;
}
function Zr(e, t, n) {
  t.time = Date.now() + n, e.ca.push(t), e.ca.sort(function(r, o) {
    return r.time - o.time;
  });
}
function Fr(e, t) {
  for (var n = 0; n < e.ca.length; n++)
    if (e.ca[n].Db == t) {
      e.ca.splice(n, 1);
      break;
    }
}
function Ye(e, t, n) {
  if (!e.P)
    throw Error("Unexpected call to createGetter");
  e.P = !1, n = Array.isArray(n) ? n[0] : n;
  var r = e.ya();
  return r.type = "CallExpression", e = new m(r, e.o[e.o.length - 1].scope), e.ia = 2, e.B = n, e.U = t, e.Qa = !0, e.F = [], e;
}
function rr(e, t, n, r) {
  if (!e.Ka)
    throw Error("Unexpected call to createSetter");
  e.Ka = !1, n = Array.isArray(n) ? n[0] : e.Na;
  var o = e.ya();
  return o.type = "CallExpression", e = new m(o, e.o[e.o.length - 1].scope), e.ia = 2, e.B = n, e.U = t, e.Qa = !0, e.F = [r], e;
}
function _t(e, t) {
  return t == null ? e.Na : t instanceof j ? t : (e = e.s(ae(e, t)), e.data = t, e);
}
b.Xb = function() {
  return this.N;
};
b.Yb = function() {
  return this.o;
};
b.ec = function(e) {
  this.o = e;
};
function m(e, t) {
  this.node = e, this.scope = t;
}
function Qr(e, t, n) {
  this.Va = e, this.S = t, this.object = n;
}
function j(e) {
  this.V = /* @__PURE__ */ Object.create(null), this.Y = /* @__PURE__ */ Object.create(null), this.h = /* @__PURE__ */ Object.create(null), this.xa = e;
}
b = j.prototype;
b.xa = null;
b.H = "Object";
b.data = null;
b.toString = function() {
  if (!Ve)
    return "[object Interpreter.Object]";
  if (!(this instanceof j))
    return String(this);
  if (this.H === "Array") {
    var e = Ur;
    e.push(this);
    try {
      var t = [], n = this.h.length, r = !1;
      1024 < n && (n = 1e3, r = !0);
      for (var o = 0; o < n; o++) {
        var s = this.h[o];
        t[o] = s instanceof j && e.indexOf(s) !== -1 ? "..." : s;
      }
      r && t.push("...");
    } finally {
      e.pop();
    }
    return t.join(",");
  }
  if (this.H === "Error") {
    if (e = Ur, e.indexOf(this) !== -1)
      return "[object Error]";
    n = this;
    do
      if ("name" in n.h) {
        t = n.h.name;
        break;
      }
    while (n = n.xa);
    n = this;
    do
      if ("message" in n.h) {
        r = n.h.message;
        break;
      }
    while (n = n.xa);
    e.push(this);
    try {
      t = t && String(t), r = r && String(r);
    } finally {
      e.pop();
    }
    return r ? t + ": " + r : String(t);
  }
  return this.data !== null ? String(this.data) : "[object " + this.H + "]";
};
b.valueOf = function() {
  return !Ve || this.data === void 0 || this.data === null || this.data instanceof RegExp ? this : this.data instanceof Date ? this.data.valueOf() : this.data;
};
function zn(e, t, n, r, o) {
  this.zb = e, this.Lb = t, this.scope = n, this.node = r, this.interval = o, this.Db = ++Yn, this.time = 0;
}
var Yn = 0;
h.prototype.stepArrayExpression = function(e, t, n) {
  n = n.elements;
  var r = t.A || 0;
  for (t.Oa ? (this.g(t.Oa, r, t.value), r++) : (t.Oa = je(this), t.Oa.h.length = n.length); r < n.length; ) {
    if (n[r])
      return t.A = r, new m(n[r], t.scope);
    r++;
  }
  e.pop(), e[e.length - 1].value = t.Oa;
};
h.prototype.stepAssignmentExpression = function(e, t, n) {
  if (!t.ja)
    return t.ja = !0, t = new m(n.left, t.scope), t.sa = !0, t;
  if (!t.Da)
    return t.Fa || (t.Fa = t.value), t.Ba && (t.ma = t.value), !t.Ba && n.operator !== "=" && (e = er(this, t.Fa), t.ma = e, this.P) ? (t.Ba = !0, Ye(this, e, t.Fa)) : (t.Da = !0, n.operator === "=" && n.left.type === "Identifier" && (t.Pa = n.left.name), new m(n.right, t.scope));
  if (t.ta)
    e.pop(), e[e.length - 1].value = t.ib;
  else {
    var r = t.ma, o = t.value;
    switch (n.operator) {
      case "=":
        r = o;
        break;
      case "+=":
        r += o;
        break;
      case "-=":
        r -= o;
        break;
      case "*=":
        r *= o;
        break;
      case "/=":
        r /= o;
        break;
      case "%=":
        r %= o;
        break;
      case "<<=":
        r <<= o;
        break;
      case ">>=":
        r >>= o;
        break;
      case ">>>=":
        r >>>= o;
        break;
      case "&=":
        r &= o;
        break;
      case "^=":
        r ^= o;
        break;
      case "|=":
        r |= o;
        break;
      default:
        throw SyntaxError("Unknown assignment expression: " + n.operator);
    }
    if (n = tr(this, t.Fa, r))
      return t.ta = !0, t.ib = r, rr(this, n, t.Fa, r);
    e.pop(), e[e.length - 1].value = r;
  }
};
h.prototype.stepBinaryExpression = function(e, t, n) {
  if (!t.ja)
    return t.ja = !0, new m(n.left, t.scope);
  if (!t.Da)
    return t.Da = !0, t.ma = t.value, new m(n.right, t.scope);
  e.pop();
  var r = t.ma;
  switch (t = t.value, n.operator) {
    case "==":
      n = r == t;
      break;
    case "!=":
      n = r != t;
      break;
    case "===":
      n = r === t;
      break;
    case "!==":
      n = r !== t;
      break;
    case ">":
      n = r > t;
      break;
    case ">=":
      n = r >= t;
      break;
    case "<":
      n = r < t;
      break;
    case "<=":
      n = r <= t;
      break;
    case "+":
      n = r + t;
      break;
    case "-":
      n = r - t;
      break;
    case "*":
      n = r * t;
      break;
    case "/":
      n = r / t;
      break;
    case "%":
      n = r % t;
      break;
    case "&":
      n = r & t;
      break;
    case "|":
      n = r | t;
      break;
    case "^":
      n = r ^ t;
      break;
    case "<<":
      n = r << t;
      break;
    case ">>":
      n = r >> t;
      break;
    case ">>>":
      n = r >>> t;
      break;
    case "in":
      t instanceof j || w(this, this.j, "'in' expects an object, not '" + t + "'"), n = ct(this, t, r);
      break;
    case "instanceof":
      G(this, t, this.O) || w(this, this.j, "'instanceof' expects an object, not '" + t + "'"), n = r instanceof j ? G(this, r, t) : !1;
      break;
    default:
      throw SyntaxError("Unknown binary operator: " + n.operator);
  }
  e[e.length - 1].value = n;
};
h.prototype.stepBlockStatement = function(e, t, n) {
  var r = t.A || 0;
  if (n = n.body[r])
    return t.A = r + 1, new m(n, t.scope);
  e.pop();
};
h.prototype.stepBreakStatement = function(e, t, n) {
  ze(this, 1, void 0, n.label && n.label.name);
};
h.prototype.Hb = 0;
h.prototype.stepCallExpression = function(e, t, n) {
  if (!t.ia) {
    t.ia = 1;
    var r = new m(n.callee, t.scope);
    return r.sa = !0, r;
  }
  if (t.ia === 1) {
    t.ia = 2;
    var o = t.value;
    if (Array.isArray(o)) {
      if (t.U = er(this, o), o[0] === Ae ? t.Pb = o[1] === "eval" : t.B = o[0], o = t.U, this.P)
        return t.ia = 1, Ye(this, o, t.value);
    } else
      t.U = o;
    t.F = [], t.A = 0;
  }
  if (o = t.U, !t.Qa) {
    if (t.A !== 0 && t.F.push(t.value), n.arguments[t.A])
      return new m(n.arguments[t.A++], t.scope);
    if (n.type === "NewExpression") {
      if (o instanceof j && !o.Ab || w(this, this.j, J(this, n.callee) + " is not a constructor"), o === this.qa)
        t.B = je(this);
      else {
        var s = o.h.prototype;
        (typeof s != "object" || s === null) && (s = this.M), t.B = this.s(s);
      }
      t.isConstructor = !0;
    }
    t.Qa = !0;
  }
  if (t.eb)
    e.pop(), e[e.length - 1].value = t.isConstructor && typeof t.value != "object" ? t.B : t.value;
  else {
    if (t.eb = !0, o instanceof j || w(this, this.j, J(this, n.callee) + " is not a function"), e = o.node) {
      for (n = $t(this, e.body, o.Va), r = je(this), o = 0; o < t.F.length; o++)
        this.g(r, o, t.F[o]);
      for (this.g(n.object, "arguments", r), o = 0; o < e.oa.length; o++)
        this.g(n.object, e.oa[o].name, t.F.length > o ? t.F[o] : void 0);
      return n.S || (t.B = _t(this, t.B)), this.g(n.object, "this", t.B, Ht), t.value = void 0, new m(e.body, n);
    }
    if (o.eval)
      if (o = t.F[0], typeof o != "string")
        t.value = o;
      else {
        try {
          r = Oe(String(o), "eval" + this.Hb++);
        } catch (f) {
          w(this, this.T, "Invalid code: " + f.message);
        }
        return o = this.ya(), o.type = "EvalProgram_", o.body = r.body, ut(o, n.start, n.end), n = t.Pb ? t.scope : this.N, n.S ? n = $t(this, r, n) : Ke(this, r, n), this.value = void 0, new m(o, n);
      }
    else if (o.Ta)
      t.scope.S || (t.B = _t(this, t.B)), t.value = o.Ta.apply(t.B, t.F);
    else if (o.Za) {
      var a = this;
      r = o.Za.length - 1, r = t.F.concat(Array(r)).slice(0, r), r.push(function(f) {
        t.value = f, a.za = !1;
      }), this.za = !0, t.scope.S || (t.B = _t(this, t.B)), o.Za.apply(t.B, r);
    } else
      w(this, this.j, J(this, n.callee) + " is not callable");
  }
};
h.prototype.stepConditionalExpression = function(e, t, n) {
  var r = t.na || 0;
  if (r === 0)
    return t.na = 1, new m(n.test, t.scope);
  if (r === 1) {
    if (t.na = 2, (r = !!t.value) && n.da)
      return new m(n.da, t.scope);
    if (!r && n.alternate)
      return new m(n.alternate, t.scope);
    this.value = void 0;
  }
  e.pop(), n.type === "ConditionalExpression" && (e[e.length - 1].value = t.value);
};
h.prototype.stepContinueStatement = function(e, t, n) {
  ze(this, 2, void 0, n.label && n.label.name);
};
h.prototype.stepDebuggerStatement = function(e) {
  e.pop();
};
h.prototype.stepDoWhileStatement = function(e, t, n) {
  if (n.type === "DoWhileStatement" && t.ga === void 0 && (t.value = !0, t.ga = !0), !t.ga)
    return t.ga = !0, new m(n.test, t.scope);
  if (!t.value)
    e.pop();
  else if (n.body)
    return t.ga = !1, t.W = !0, new m(n.body, t.scope);
};
h.prototype.stepEmptyStatement = function(e) {
  e.pop();
};
h.prototype.stepEvalProgram_ = function(e, t, n) {
  var r = t.A || 0;
  if (n = n.body[r])
    return t.A = r + 1, new m(n, t.scope);
  e.pop(), e[e.length - 1].value = this.value;
};
h.prototype.stepExpressionStatement = function(e, t, n) {
  if (!t.ka)
    return this.value = void 0, t.ka = !0, new m(n.la, t.scope);
  e.pop(), this.value = t.value;
};
h.prototype.stepForInStatement = function(e, t, n) {
  if (!t.Ub && (t.Ub = !0, n.left.fa && n.left.fa[0].ua))
    return t.scope.S && w(this, this.T, "for-in loop variable declaration may not have an initializer"), new m(n.left, t.scope);
  if (!t.Ca)
    return t.Ca = !0, t.pa || (t.pa = t.value), new m(n.right, t.scope);
  if (t.W || (t.W = !0, t.u = t.value, t.kb = /* @__PURE__ */ Object.create(null)), t.Ra === void 0)
    e:
      for (; ; ) {
        if (t.u instanceof j)
          for (t.wa || (t.wa = Object.getOwnPropertyNames(t.u.h)); ; ) {
            var r = t.wa.shift();
            if (r === void 0)
              break;
            if (Object.prototype.hasOwnProperty.call(t.u.h, r) && !t.kb[r] && (t.kb[r] = !0, Object.prototype.propertyIsEnumerable.call(t.u.h, r))) {
              t.Ra = r;
              break e;
            }
          }
        else if (t.u !== null && t.u !== void 0) {
          for (t.wa || (t.wa = Object.getOwnPropertyNames(t.u)); r = t.wa.shift(), r !== void 0; )
            if (t.kb[r] = !0, Object.prototype.propertyIsEnumerable.call(t.u, r)) {
              t.Ra = r;
              break e;
            }
        }
        if (t.u = ae(this, t.u), t.wa = null, t.u === null) {
          e.pop();
          return;
        }
      }
  if (!t.wb)
    if (t.wb = !0, e = n.left, e.type === "VariableDeclaration")
      t.pa = [Ae, e.fa[0].id.name];
    else
      return t.pa = null, t = new m(e, t.scope), t.sa = !0, t;
  if (t.pa || (t.pa = t.value), !t.ta && (t.ta = !0, e = t.Ra, r = tr(this, t.pa, e)))
    return rr(this, r, t.pa, e);
  if (t.Ra = void 0, t.wb = !1, t.ta = !1, n.body)
    return new m(n.body, t.scope);
};
h.prototype.stepForStatement = function(e, t, n) {
  switch (t.na) {
    default:
      if (t.na = 1, n.ua)
        return new m(n.ua, t.scope);
      break;
    case 1:
      if (t.na = 2, n.test)
        return new m(n.test, t.scope);
      break;
    case 2:
      if (t.na = 3, n.test && !t.value)
        e.pop();
      else
        return t.W = !0, new m(n.body, t.scope);
      break;
    case 3:
      if (t.na = 1, n.update)
        return new m(n.update, t.scope);
  }
};
h.prototype.stepFunctionDeclaration = function(e) {
  e.pop();
};
h.prototype.stepFunctionExpression = function(e, t, n) {
  e.pop(), t = e[e.length - 1], e = t.scope, n.id && (e = Qt(this, e)), t.value = Yt(this, n, e, t.Pa), n.id && this.g(e.object, n.id.name, t.value, Ht);
};
h.prototype.stepIdentifier = function(e, t, n) {
  if (e.pop(), t.sa)
    e[e.length - 1].value = [Ae, n.name];
  else {
    if (t = qt(this, n.name), this.P)
      return Ye(this, t, this.Na);
    e[e.length - 1].value = t;
  }
};
h.prototype.stepIfStatement = h.prototype.stepConditionalExpression;
h.prototype.stepLabeledStatement = function(e, t, n) {
  return e.pop(), e = t.labels || [], e.push(n.label.name), t = new m(n.body, t.scope), t.labels = e, t;
};
h.prototype.stepLiteral = function(e, t, n) {
  e.pop(), t = n.value, t instanceof RegExp && (n = this.s(this.Ma), Kt(this, n, t), t = n), e[e.length - 1].value = t;
};
h.prototype.stepLogicalExpression = function(e, t, n) {
  if (n.operator !== "&&" && n.operator !== "||")
    throw SyntaxError("Unknown logical operator: " + n.operator);
  if (!t.ja)
    return t.ja = !0, new m(n.left, t.scope);
  if (t.Da)
    e.pop(), e[e.length - 1].value = t.value;
  else if (n.operator === "&&" && !t.value || n.operator === "||" && t.value)
    e.pop(), e[e.length - 1].value = t.value;
  else
    return t.Da = !0, new m(n.right, t.scope);
};
h.prototype.stepMemberExpression = function(e, t, n) {
  if (!t.Ca)
    return t.Ca = !0, new m(n.object, t.scope);
  if (n.bb)
    if (t.Vb)
      n = t.value;
    else
      return t.u = t.value, t.Vb = !0, new m(n.Wa, t.scope);
  else
    t.u = t.value, n = n.Wa.name;
  if (e.pop(), t.sa)
    e[e.length - 1].value = [t.u, n];
  else {
    if (n = this.G(t.u, n), this.P)
      return Ye(this, n, t.u);
    e[e.length - 1].value = n;
  }
};
h.prototype.stepNewExpression = h.prototype.stepCallExpression;
h.prototype.stepObjectExpression = function(e, t, n) {
  var r = t.A || 0, o = n.h[r];
  if (t.u) {
    var s = t.Pa;
    t.Ja[s] || (t.Ja[s] = {}), t.Ja[s][o.kind] = t.value, t.A = ++r, o = n.h[r];
  } else
    t.u = this.s(this.M), t.Ja = /* @__PURE__ */ Object.create(null);
  if (o) {
    var a = o.key;
    if (a.type === "Identifier")
      s = a.name;
    else if (a.type === "Literal")
      s = a.value;
    else
      throw SyntaxError("Unknown object structure: " + a.type);
    return t.Pa = s, new m(o.value, t.scope);
  }
  for (a in t.Ja)
    n = t.Ja[a], "get" in n || "set" in n ? this.g(t.u, a, Je, {
      configurable: !0,
      enumerable: !0,
      get: n.get,
      set: n.set
    }) : this.g(t.u, a, n.init);
  e.pop(), e[e.length - 1].value = t.u;
};
h.prototype.stepProgram = function(e, t, n) {
  if (e = n.body.shift())
    return t.done = !1, new m(e, t.scope);
  t.done = !0;
};
h.prototype.stepReturnStatement = function(e, t, n) {
  if (n.K && !t.ka)
    return t.ka = !0, new m(n.K, t.scope);
  ze(this, 3, t.value);
};
h.prototype.stepSequenceExpression = function(e, t, n) {
  var r = t.A || 0;
  if (n = n.xb[r])
    return t.A = r + 1, new m(n, t.scope);
  e.pop(), e[e.length - 1].value = t.value;
};
h.prototype.stepSwitchStatement = function(e, t, n) {
  if (!t.ga)
    return t.ga = 1, new m(n.Qb, t.scope);
  for (t.ga === 1 && (t.ga = 2, t.hc = t.value, t.cb = -1); ; ) {
    var r = t.gb || 0, o = n.tb[r];
    if (t.Ha || !o || o.test)
      if (o || t.Ha || t.cb === -1)
        if (o) {
          if (!t.Ha && !t.Fb && o.test)
            return t.Fb = !0, new m(o.test, t.scope);
          if (t.Ha || t.value === t.hc) {
            t.Ha = !0;
            var s = t.A || 0;
            if (o.da[s])
              return t.Zb = !0, t.A = s + 1, new m(o.da[s], t.scope);
          }
          t.Fb = !1, t.A = 0, t.gb = r + 1;
        } else {
          e.pop();
          break;
        }
      else
        t.Ha = !0, t.gb = t.cb;
    else
      t.cb = r, t.gb = r + 1;
  }
};
h.prototype.stepThisExpression = function(e) {
  e.pop(), e[e.length - 1].value = qt(this, "this");
};
h.prototype.stepThrowStatement = function(e, t, n) {
  if (t.ka)
    w(this, t.value);
  else
    return t.ka = !0, new m(n.K, t.scope);
};
h.prototype.stepTryStatement = function(e, t, n) {
  if (!t.Rb)
    return t.Rb = !0, new m(n.block, t.scope);
  if (t.ea && t.ea.type === 4 && !t.Tb && n.Ea)
    return t.Tb = !0, e = Qt(this, t.scope), this.g(e.object, n.Ea.Ua.name, t.ea.value), t.ea = void 0, new m(n.Ea.body, e);
  if (!t.Sb && n.fb)
    return t.Sb = !0, new m(n.fb, t.scope);
  e.pop(), t.ea && ze(this, t.ea.type, t.ea.value, t.ea.label);
};
h.prototype.stepUnaryExpression = function(e, t, n) {
  if (!t.ka)
    return t.ka = !0, e = new m(n.K, t.scope), e.sa = n.operator === "delete", e;
  e.pop();
  var r = t.value;
  switch (n.operator) {
    case "-":
      r = -r;
      break;
    case "+":
      r = +r;
      break;
    case "!":
      r = !r;
      break;
    case "~":
      r = ~r;
      break;
    case "delete":
      if (n = !0, Array.isArray(r)) {
        var o = r[0];
        o === Ae && (o = t.scope), r = String(r[1]);
        try {
          delete o.h[r];
        } catch {
          t.scope.S ? w(this, this.j, "Cannot delete property '" + r + "' of '" + o + "'") : n = !1;
        }
      }
      r = n;
      break;
    case "typeof":
      r = r && r.H === "Function" ? "function" : typeof r;
      break;
    case "void":
      r = void 0;
      break;
    default:
      throw SyntaxError("Unknown unary operator: " + n.operator);
  }
  e[e.length - 1].value = r;
};
h.prototype.stepUpdateExpression = function(e, t, n) {
  if (!t.ja)
    return t.ja = !0, e = new m(n.K, t.scope), e.sa = !0, e;
  if (t.Ga || (t.Ga = t.value), t.Ba && (t.ma = t.value), !t.Ba) {
    var r = er(this, t.Ga);
    if (t.ma = r, this.P)
      return t.Ba = !0, Ye(this, r, t.Ga);
  }
  if (t.ta)
    e.pop(), e[e.length - 1].value = t.ib;
  else {
    if (r = Number(t.ma), n.operator === "++")
      var o = r + 1;
    else if (n.operator === "--")
      o = r - 1;
    else
      throw SyntaxError("Unknown update expression: " + n.operator);
    if (n = n.prefix ? o : r, r = tr(this, t.Ga, o))
      return t.ta = !0, t.ib = n, rr(this, r, t.Ga, o);
    e.pop(), e[e.length - 1].value = n;
  }
};
h.prototype.stepVariableDeclaration = function(e, t, n) {
  n = n.fa;
  var r = t.A || 0, o = n[r];
  for (t.Bb && o && (Yr(this, o.id.name, t.value), t.Bb = !1, o = n[++r]); o; ) {
    if (o.ua)
      return t.A = r, t.Bb = !0, t.Pa = o.id.name, new m(o.ua, t.scope);
    o = n[++r];
  }
  e.pop();
};
h.prototype.stepWithStatement = function(e, t, n) {
  return t.Ca ? (e.pop(), e = Qt(this, t.scope, t.value), new m(n.body, e)) : (t.Ca = !0, new m(n.object, t.scope));
};
h.prototype.stepWhileStatement = h.prototype.stepDoWhileStatement;
pe.Interpreter = h;
h.prototype.step = h.prototype.lb;
h.prototype.run = h.prototype.Eb;
h.prototype.appendCode = h.prototype.Jb;
h.prototype.createObject = h.prototype.Aa;
h.prototype.createObjectProto = h.prototype.s;
h.prototype.createAsyncFunction = h.prototype.ub;
h.prototype.createNativeFunction = h.prototype.i;
h.prototype.getProperty = h.prototype.G;
h.prototype.setProperty = h.prototype.g;
h.prototype.nativeToPseudo = h.prototype.Ia;
h.prototype.pseudoToNative = h.prototype.R;
h.prototype.getGlobalScope = h.prototype.Xb;
h.prototype.getStateStack = h.prototype.Yb;
h.prototype.setStateStack = h.prototype.ec;
h.VALUE_IN_DESCRIPTOR = Je;
function Zn() {
  const e = h.prototype.getProperty, t = h.prototype.setProperty;
  function n(a, f) {
    if (a == null || !a._connected)
      return e.call(this, a, f);
    const d = a._connected[f];
    return Array.isArray(d) ? this.nativeToPseudo(d) : typeof d == "object" ? this.createConnectedObject(d) : d;
  }
  function r(a, f, d, v) {
    if (a == null || !a._connected)
      return t.call(this, a, f, d, v);
    a._connected[f] = this.pseudoToNative(d);
  }
  const o = [], s = [];
  for (const a of Object.keys(h.prototype))
    h.prototype[a] === e && o.push(a), h.prototype[a] === t && s.push(a);
  for (const a of o)
    h.prototype[a] = n;
  for (const a of s)
    h.prototype[a] = r;
  h.prototype.createConnectedObject = function(a) {
    const f = this.createObject(this.OBJECT);
    return f._connected = a, f;
  };
}
Zn();
const Qn = (e) => e.replace(/^.*?function main\(\)/, `
var __awaiter = function (e, t, n, r) {return r()},
__generator = function (e, t) { return t() };
function main()`).replace(/\?\./g, "."), Wr = (e) => e + "JSON", qn = ({
  builder: e,
  context: t,
  event: n,
  rootState: r,
  localState: o,
  rootSetState: s,
  code: a
}) => {
  const f = Xt({
    rootState: r,
    localState: o,
    rootSetState: s
  }), d = Jr({
    builder: e,
    context: t,
    event: n,
    state: f
  }), v = d.map(([O]) => {
    const R = Wr(O);
    return O === "state" ? "" : `var ${O} = ${R} === undefined ? undefined : JSON.parse(${R});`;
  }).join(`
`), I = Qn(a);
  if (I === "")
    return;
  const A = `
function theFunction() {
  ${v}

  ${I}
}
theFunction();
`, W = function(O, R) {
    d.forEach(([_, k]) => {
      if (_ === "state")
        O.setProperty(R, _, O.createConnectedObject(k), O.READONLY_DESCRIPTOR);
      else {
        const ue = JSON.stringify(k);
        O.setProperty(R, Wr(_), ue);
      }
    });
  }, V = new h(A, W);
  return V.run(), V.pseudoToNative(V.value);
};
function eo() {
  var e;
  return typeof process != "undefined" && st((e = process == null ? void 0 : process.versions) == null ? void 0 : e.node);
}
const to = ({
  shouldLogWarning: e
}) => {
  var o;
  if (!eo())
    return !1;
  const t = process.arch === "arm64", n = process.version.startsWith("v20"), r = (o = process.env.NODE_OPTIONS) == null ? void 0 : o.includes("--no-node-snapshot");
  return t && n && !r ? (e && Hr.log("Skipping usage of `isolated-vm` to avoid crashes in Node v20 on an arm64 machine.\n    If you would like to use the `isolated-vm` package on this machine, please provide the `NODE_OPTIONS=--no-node-snapshot` config to your Node process.\n    See https://github.com/BuilderIO/builder/blob/main/packages/sdks/README.md#node-v20--m1-macs-apple-silicon-support for more information.\n    "), !0) : !1;
}, ro = (e) => Ft() || to({
  shouldLogWarning: !0
}) ? Bn(e) : qn(e), no = /^(return )?(\s*)?state(?<getPath>(\.\w+)+)(\s*);?$/, oo = /(\s)*var(\s)+_virtual_index(\s)*=(\s)*state(?<getPath>(\.\w+)+)(\s*);?(\s)*return(\s)*_virtual_index(\s)*/, io = (e) => {
  var t, n, r, o, s, a;
  return ((r = (n = (t = no.exec(e.trim())) == null ? void 0 : t.groups) == null ? void 0 : n.getPath) == null ? void 0 : r.slice(1)) || ((a = (s = (o = oo.exec(e.trim())) == null ? void 0 : o.groups) == null ? void 0 : s.getPath) == null ? void 0 : a.slice(1));
};
function ft({
  code: e,
  context: t,
  localState: n,
  rootState: r,
  rootSetState: o,
  event: s,
  isExpression: a = !0
}) {
  if (e.trim() === "")
    return;
  const f = io(e.trim());
  if (f)
    return bn({
      ...r,
      ...n
    }, f);
  const d = {
    code: Rn(e, {
      isExpression: a
    }),
    builder: Nn(),
    context: t,
    event: s,
    rootSetState: o,
    rootState: r,
    localState: n
  };
  try {
    return ro(d);
  } catch (v) {
    Hr.error("Failed code evaluation: " + v.message, {
      code: e
    });
    return;
  }
}
function so(e, t) {
  var n;
  return {
    ...(n = e.component) == null ? void 0 : n.options,
    ...e.options,
    ...ao(e, t)
  };
}
const ao = (e, t) => {
  var n, r;
  if (((n = e.component) == null ? void 0 : n.name) === "Text" && ((r = e.component.options) != null && r.text) && typeof e.component.options.text == "string")
    return {
      ...e.component.options,
      text: e.component.options.text.replace(/{{([^}]+)}}/g, (o, s) => ft({
        code: s,
        context: t,
        localState: t.localState,
        rootState: t.rootState,
        rootSetState: t.rootSetState
      }))
    };
};
function lo(e, ...t) {
  const n = Object.assign({}, e);
  for (const r of t)
    delete n[r];
  return n;
}
function at(e, t, n = null, r = null, o = /* @__PURE__ */ new WeakSet()) {
  if (e == null || typeof e != "object") {
    t(e, (s) => {
      n !== null && r !== null && (n[r] = s);
    });
    return;
  }
  o.has(e) || (o.add(e), Array.isArray(e) ? e.forEach((s, a) => {
    t(s, (d) => {
      e[a] = d;
    }), at(s, t, e, a, o);
  }) : Object.entries(e).forEach(([s, a]) => {
    t(a, (d) => {
      e[s] = d;
    }), at(a, t, e, s, o);
  }));
}
function qr(e) {
  return e && typeof e == "object" && e["@type"] === "@builder.io/core:LocalizedValue";
}
function uo(e) {
  if (!e || !Object.getOwnPropertyNames(e).length)
    return !1;
  let t = !1;
  return at(e, (n) => {
    if (qr(n)) {
      t = !0;
      return;
    }
  }), t;
}
function co(e, t) {
  return !e || !Object.getOwnPropertyNames(e).length ? {} : (at(e, (n, r) => {
    var o;
    qr(n) && r((o = n[t]) != null ? o : void 0);
  }), e);
}
function fo(e, t) {
  var n, r;
  return (n = e.component) != null && n.options && uo((r = e.component) == null ? void 0 : r.options) && (t || console.warn("[Builder.io] In order to use localized fields in Builder, you must pass a locale prop to the BuilderComponent or to options object while fetching the content to resolve localized fields. Learn more: https://www.builder.io/c/docs/localization-inline#targeting-and-inline-localization"), e.component.options = co(e.component.options, t != null ? t : "Default")), e;
}
const en = (e) => JSON.parse(JSON.stringify(e)), po = (e, t, n) => {
  if (Object(e) !== e)
    return e;
  const r = Array.isArray(t) ? t : t.toString().match(/[^.[\]]+/g);
  return r.slice(0, -1).reduce((o, s, a) => Object(o[s]) === o[s] ? o[s] : o[s] = Math.abs(Number(r[a + 1])) >> 0 === +r[a + 1] ? [] : {}, e)[r[r.length - 1]] = n, e;
};
function Vt(e) {
  if (e === null || typeof e != "object")
    return e;
  if (Array.isArray(e))
    return e.map((n) => Vt(n));
  if (e["@type"] === "@builder.io/sdk:Element")
    return e;
  const t = {};
  for (const n in e)
    n !== "meta" && Object.prototype.hasOwnProperty.call(e, n) && (t[n] = Vt(e[n]));
  return t;
}
const ho = ["svelte", "vue", "angular", "qwik", "solid"].includes(lt), go = (e) => {
  if (ho) {
    const t = en(e);
    return {
      ...t,
      properties: {
        ...t.properties
      },
      actions: {
        ...t.actions
      }
    };
  } else {
    const t = Vt(lo(e, "children", "meta"));
    return {
      ...t,
      properties: {
        ...t.properties
      },
      actions: {
        ...t.actions
      },
      children: e.children,
      meta: e.meta
    };
  }
}, yo = ({
  block: e,
  context: t,
  localState: n,
  rootState: r,
  rootSetState: o
}) => {
  if (!e.bindings)
    return e;
  const s = go(e);
  for (const a in e.bindings) {
    const f = e.bindings[a], d = ft({
      code: f,
      localState: n,
      rootState: r,
      rootSetState: o,
      context: t
    });
    po(s, a, d);
  }
  return s;
};
function mo({
  block: e,
  context: t,
  localState: n,
  rootState: r,
  rootSetState: o
}) {
  let s = e;
  return s = yo({
    block: s,
    localState: n,
    rootState: r,
    rootSetState: o,
    context: t
  }), s = fo(s, r.locale), s;
}
const pt = (e) => e ? e.replace(/([a-z0-9]|(?=[A-Z]))([A-Z])/g, "$1-$2").toLowerCase() : "";
function vo(e, t, n = {}) {
  let r, o, s, a = null, f = 0;
  const d = function() {
    f = n.leading === !1 ? 0 : Date.now(), a = null, s = e.apply(r, o), a || (r = o = null);
  };
  return function() {
    const v = Date.now();
    !f && n.leading === !1 && (f = v);
    const I = t - (v - f);
    return r = this, o = arguments, I <= 0 || I > t ? (a && (clearTimeout(a), a = null), f = v, s = e.apply(r, o), a || (r = o = null)) : !a && n.trailing !== !1 && (a = setTimeout(d, I)), s;
  };
}
function He(e, ...t) {
  const n = Object(e);
  for (let r = 1; r < arguments.length; r++) {
    const o = arguments[r];
    if (o != null)
      for (const s in o)
        Object.prototype.hasOwnProperty.call(o, s) && (n[s] = o[s]);
  }
  return n;
}
function wo(e) {
  for (const t of e)
    switch (t.trigger) {
      case "pageLoad":
        So(t);
        break;
      case "scrollInView":
        Eo(t);
        break;
    }
}
function tn(e) {
  console.warn(`Cannot animate element: element with ID ${e} not found!`);
}
function rn(e, t) {
  const n = xo(e), r = getComputedStyle(t), o = e.steps[0].styles, s = e.steps[e.steps.length - 1].styles, a = [o, s];
  for (const f of a)
    for (const d of n)
      d in f || (f[d] = r[d]);
}
function xo(e) {
  const t = [];
  for (const n of e.steps)
    for (const r in n.styles)
      t.indexOf(r) === -1 && t.push(r);
  return t;
}
function So(e) {
  const t = Array.prototype.slice.call(document.getElementsByClassName(e.elementId || e.id || ""));
  if (!t.length) {
    tn(e.elementId || e.id || "");
    return;
  }
  Array.from(t).forEach((n) => {
    rn(e, n), n.style.transition = "none", n.style.transitionDelay = "0", He(n.style, e.steps[0].styles), setTimeout(() => {
      n.style.transition = `all ${e.duration}s ${pt(e.easing)}`, e.delay && (n.style.transitionDelay = e.delay + "s"), He(n.style, e.steps[1].styles), setTimeout(() => {
        n.style.transition = "", n.style.transitionDelay = "";
      }, (e.delay || 0) * 1e3 + e.duration * 1e3 + 100);
    });
  });
}
function Eo(e) {
  const t = Array.prototype.slice.call(document.getElementsByClassName(e.elementId || e.id || ""));
  if (!t.length) {
    tn(e.elementId || e.id || "");
    return;
  }
  Array.from(t).forEach((n) => {
    rn(e, n);
    let r = !1, o = !1;
    function s() {
      !r && f(n) ? (r = !0, o = !0, setTimeout(() => {
        He(n.style, e.steps[1].styles), e.repeat || document.removeEventListener("scroll", a), setTimeout(() => {
          o = !1, e.repeat || (n.style.transition = "", n.style.transitionDelay = "");
        }, (e.duration + (e.delay || 0)) * 1e3 + 100);
      })) : e.repeat && r && !o && !f(n) && (r = !1, He(n.style, e.steps[0].styles));
    }
    const a = vo(s, 200, {
      leading: !1
    });
    function f(I) {
      const A = I.getBoundingClientRect(), W = window.innerHeight, P = (e.thresholdPercent || 0) / 100 * W;
      return A.bottom > P && A.top < W - P;
    }
    const d = e.steps[0].styles;
    function v() {
      He(n.style, d);
    }
    v(), setTimeout(() => {
      n.style.transition = `all ${e.duration}s ${pt(e.easing)}`, e.delay && (n.style.transitionDelay = e.delay + "s");
    }), document.addEventListener("scroll", a, {
      capture: !0,
      passive: !0
    }), s();
  });
}
const nn = (e) => Object.entries(e).map(([n, r]) => {
  if (typeof r == "string")
    return `${pt(n)}: ${r};`;
}).filter(st), ko = (e) => nn(e).join(`
`), Ue = ({
  mediaQuery: e,
  className: t,
  styles: n
}) => {
  const r = `.${t} {
    ${ko(n)}
  }`;
  return e ? `${e} {
      ${r}
    }` : r;
};
function Co({
  style: e
}) {
  return e;
}
const Oo = ({
  block: e,
  context: t
}) => jo(Co({
  style: e.style || {},
  context: t,
  block: e
}));
function jo(e) {
  switch (lt) {
    case "svelte":
    case "vue":
    case "solid":
    case "angular":
      return nn(e).join(" ");
    case "qwik":
    case "reactNative":
    case "react":
    case "rsc":
      return e;
  }
}
const on = (e, t) => e ? t ? e.models && e.models.length > 0 && !e.models.includes(t) : !1 : !0, Ao = ({
  block: e,
  registeredComponents: t,
  model: n
}) => {
  var s;
  const r = (s = e.component) == null ? void 0 : s.name;
  if (!r)
    return null;
  const o = t[r];
  if (!o || on(o, n)) {
    console.warn(`
      Could not find a registered component named "${r}". 
      If you registered it, is the file that registered it imported by the file that needs to render it?`);
    return;
  } else
    return o;
}, Io = ({
  block: e,
  context: t
}) => {
  const {
    repeat: n,
    ...r
  } = e;
  if (!(n != null && n.collection))
    return;
  const o = ft({
    code: n.collection,
    localState: t.localState,
    rootState: t.rootState,
    rootSetState: t.rootSetState,
    context: t.context
  });
  if (!Array.isArray(o))
    return;
  const s = n.collection.split(".").pop(), a = n.itemName || (s ? s + "Item" : "item");
  return o.map((d, v) => ({
    context: {
      ...t,
      localState: {
        ...t.localState,
        $index: v,
        $item: d,
        [a]: d,
        [`$${a}Index`]: v
      }
    },
    block: r
  }));
}, To = (e, t) => {
  var n;
  return (n = e == null ? void 0 : e.shouldReceiveBuilderProps) != null && n.builderLinkComponent ? {
    builderLinkComponent: t
  } : {};
}, Po = (e, t, n) => {
  var r;
  return (r = e == null ? void 0 : e.shouldReceiveBuilderProps) != null && r.builderComponents ? {
    builderComponents: Object.fromEntries(Object.entries(t).filter(([s, a]) => !on(a, n)))
  } : {};
}, bo = (e, t) => {
  var n;
  return (n = e == null ? void 0 : e.shouldReceiveBuilderProps) != null && n.builderBlock ? {
    builderBlock: t
  } : {};
}, No = (e, t) => {
  var n;
  return (n = e == null ? void 0 : e.shouldReceiveBuilderProps) != null && n.builderContext ? {
    builderContext: t
  } : {};
}, Gr = (e) => e.toString(), sn = {
  xsmall: {
    min: 0,
    default: 160,
    max: 320
  },
  small: {
    min: 321,
    default: 321,
    max: 640
  },
  medium: {
    min: 641,
    default: 642,
    max: 991
  },
  large: {
    min: 990,
    default: 991,
    max: 1200
  }
}, Ut = (e, t = sn) => `@media (max-width: ${t[e].max}px)`, Ro = (e) => {
  const t = en(sn);
  if (!e)
    return t;
  const {
    xsmall: n,
    small: r,
    medium: o
  } = e;
  if (n) {
    const d = Math.floor(n / 2);
    t.xsmall = {
      max: n,
      min: d,
      default: d + 1
    };
  }
  if (!r || !o)
    return t;
  const s = n ? t.xsmall.max + 1 : Math.floor(r / 2);
  t.small = {
    max: r,
    min: s,
    default: s + 1
  };
  const a = t.small.max + 1;
  t.medium = {
    max: o,
    min: a,
    default: a + 1
  };
  const f = t.medium.max + 1;
  return t.large = {
    max: 2e3,
    // TODO: decide upper limit
    min: f,
    default: f + 1
  }, t;
};
function Bo(e) {
  return /* @__PURE__ */ T(
    "style",
    {
      dangerouslySetInnerHTML: { __html: e.styles },
      "data-id": e.id,
      nonce: e.nonce
    }
  );
}
function Mo(e) {
  function t() {
    const r = e.block;
    return st(r.hide) ? !r.hide : st(r.show) ? r.show : !0;
  }
  function n() {
    var ue, Ie, Te, B, Z;
    const r = e.block, o = r.responsiveStyles, s = e.context.content, a = Ro(
      ((ue = s == null ? void 0 : s.meta) == null ? void 0 : ue.breakpoints) || {}
    ), f = !!((Te = (Ie = s == null ? void 0 : s.meta) == null ? void 0 : Ie.breakpoints) != null && Te.xsmall), d = o == null ? void 0 : o.large, v = o == null ? void 0 : o.medium, I = o == null ? void 0 : o.small, A = o == null ? void 0 : o.xsmall, W = r.id;
    if (!W)
      return "";
    const V = d ? Ue({
      className: W,
      styles: d
    }) : "", P = v ? Ue({
      className: W,
      styles: v,
      mediaQuery: Ut(
        "medium",
        a
      )
    }) : "", O = I ? Ue({
      className: W,
      styles: I,
      mediaQuery: Ut(
        "small",
        a
      )
    }) : "", R = A && f ? Ue({
      className: W,
      styles: A,
      mediaQuery: Ut(
        "xsmall",
        a
      )
    }) : "", _ = r.animations && r.animations.find((x) => x.trigger === "hover");
    let k = "";
    if (_) {
      const x = ((Z = (B = _.steps) == null ? void 0 : B[1]) == null ? void 0 : Z.styles) || {};
      k = Ue({
        className: `${W}:hover`,
        styles: {
          ...x,
          transition: `all ${_.duration}s ${pt(
            _.easing
          )}`,
          transitionDelay: _.delay ? `${_.delay}s` : "0s"
        }
      }) || "";
    }
    return [
      V,
      P,
      O,
      R,
      k
    ].join(" ");
  }
  return /* @__PURE__ */ T(X, { children: n() && t() ? /* @__PURE__ */ T(X, { children: /* @__PURE__ */ T(
    Bo,
    {
      id: "builderio-block",
      styles: n(),
      nonce: e.context.nonce
    }
  ) }) : null });
}
function Do(e) {
  return e.charAt(0).toUpperCase() + e.slice(1);
}
const _o = (e) => `on${Do(e)}`, Uo = (e, t) => (n) => ft({
  code: e,
  context: t.context,
  localState: t.localState,
  rootState: t.rootState,
  rootSetState: t.rootSetState,
  event: n,
  isExpression: !1
});
function an(e) {
  var r;
  const t = {}, n = (r = e.block.actions) != null ? r : {};
  for (const o in n) {
    if (!n.hasOwnProperty(o))
      continue;
    const s = n[o];
    let a = _o(o);
    if (e.stripPrefix)
      switch (lt) {
        case "vue":
          a = a.replace("v-on:", "");
          break;
        case "svelte":
          a = a.replace("on:", "");
          break;
      }
    t[a] = Uo(s, e);
  }
  return t;
}
function Lo({
  properties: e
}) {
  return e;
}
const Fo = (e) => ({
  href: e.href
});
function nr({
  block: e,
  context: t
}) {
  var r;
  const n = {
    ...Fo(e),
    ...e.properties,
    "builder-id": e.id,
    style: Oo({
      block: e,
      context: t
    }),
    [In()]: [e.id, "builder-block", e.class, (r = e.properties) == null ? void 0 : r.class].filter(Boolean).join(" ")
  };
  return Lo({
    properties: n,
    context: t,
    block: e
  });
}
function Wo(e) {
  return /* @__PURE__ */ T(
    Tn,
    {
      TagName: e.Wrapper,
      attributes: nr({
        block: e.block,
        context: e.context
      }),
      actionAttributes: an({
        block: e.block,
        rootState: e.context.rootState,
        rootSetState: e.context.rootSetState,
        localState: e.context.localState,
        context: e.context.context,
        stripPrefix: !0
      }),
      children: e.children
    }
  );
}
function Go(e) {
  return ye(() => {
  }, []), /* @__PURE__ */ T(X, { children: e.children });
}
function $o(e) {
  var r, o;
  Ce(() => 0);
  function t() {
    return e.includeBlockProps ? {
      ...nr({
        block: e.block,
        context: e.context
      }),
      ...an({
        block: e.block,
        rootState: e.context.rootState,
        rootSetState: e.context.rootSetState,
        localState: e.context.localState,
        context: e.context.context
      })
    } : {};
  }
  function n() {
    return e.wrapperProps;
  }
  return ye(() => {
  }, [e.wrapperProps, (o = (r = e.block) == null ? void 0 : r.component) == null ? void 0 : o.options]), /* @__PURE__ */ T(X, { children: e.Wrapper.load ? /* @__PURE__ */ T(X, { children: /* @__PURE__ */ T(
    Go,
    {
      load: e.Wrapper.load,
      fallback: e.Wrapper.fallback,
      props: e.wrapperProps,
      attributes: t(),
      children: e.children
    }
  ) }) : /* @__PURE__ */ T(e.Wrapper, { ...n(), attributes: t(), children: e.children }) });
}
const Vo = ({
  componentOptions: e,
  builderBlock: t,
  context: n,
  componentRef: r,
  includeBlockProps: o,
  isInteractive: s,
  contextValue: a
}) => {
  const f = {
    ...e,
    /**
     * If `noWrap` is set to `true`, then the block's props/attributes are provided to the
     * component itself directly. Otherwise, they are provided to the wrapper element.
     */
    ...o ? {
      attributes: nr({
        block: t,
        context: a
      })
    } : {}
  };
  return s ? {
    Wrapper: r,
    block: t,
    context: n,
    wrapperProps: e,
    includeBlockProps: o
  } : f;
};
function $r(e) {
  var r;
  Ce(() => !1);
  const [t, n] = Ce(
    () => e.isInteractive ? $o : e.componentRef
  );
  return /* @__PURE__ */ T(X, { children: e.componentRef ? /* @__PURE__ */ T(X, { children: /* @__PURE__ */ T(
    t,
    {
      ...Vo({
        componentOptions: e.componentOptions,
        builderBlock: e.builderBlock,
        context: e.context,
        componentRef: e.componentRef,
        linkComponent: e.linkComponent,
        includeBlockProps: e.includeBlockProps,
        isInteractive: e.isInteractive,
        contextValue: e.context
      }),
      children: e.componentRef ? /* @__PURE__ */ T(X, { children: (r = e.blockChildren) == null ? void 0 : r.map((o) => /* @__PURE__ */ T(
        ht,
        {
          block: o,
          context: e.context,
          registeredComponents: e.registeredComponents,
          linkComponent: e.linkComponent
        },
        o.id
      )) }) : null
    }
  ) }) : null });
}
function Vr(e) {
  const [t, n] = Ce(() => e.repeatContext);
  return ye(() => {
    n(e.repeatContext);
  }, [e.repeatContext]), /* @__PURE__ */ T(Xr.Provider, { value: t, children: /* @__PURE__ */ T(
    ht,
    {
      block: e.block,
      context: t,
      registeredComponents: e.registeredComponents,
      linkComponent: e.linkComponent
    }
  ) });
}
function ht(e) {
  var I, A, W, V;
  function t() {
    return Io({
      block: e.block,
      context: e.context
    });
  }
  const [n, r] = Ce(() => ({
    value: null,
    update: !1
  }));
  function o() {
    var O;
    if (n.value && !n.update && !An())
      return n.value;
    const P = (O = e.block.repeat) != null && O.collection ? e.block : mo({
      block: e.block,
      localState: e.context.localState,
      rootState: e.context.rootState,
      rootSetState: e.context.rootSetState,
      context: e.context.context
    });
    return n.value = P, n.update = !1, P;
  }
  function s() {
    return Ao({
      block: o(),
      registeredComponents: e.registeredComponents,
      model: e.context.model
    });
  }
  function a() {
    var O;
    return e.block.tagName === "a" || ((O = o().properties) == null ? void 0 : O.href) || o().href ? e.linkComponent || "a" : e.block.tagName || "div";
  }
  function f() {
    var R, _;
    if ((R = e.block.repeat) != null && R.collection)
      return !!((_ = t == null ? void 0 : t()) != null && _.length);
    const P = "hide" in o() ? o().hide : !1;
    return ("show" in o() ? o().show : !0) && !P;
  }
  function d() {
    var O, R;
    return !((O = s == null ? void 0 : s()) != null && O.component) && !t() ? (R = o().children) != null ? R : [] : [];
  }
  function v() {
    var P, O, R, _;
    return {
      blockChildren: (P = o().children) != null ? P : [],
      componentRef: (O = s == null ? void 0 : s()) == null ? void 0 : O.component,
      componentOptions: {
        ...so(o(), e.context),
        ...bo(s(), o()),
        ...No(s(), e.context),
        ...To(s(), e.linkComponent),
        ...Po(
          s(),
          e.registeredComponents,
          e.context.model
        )
      },
      context: e.context,
      linkComponent: e.linkComponent,
      registeredComponents: e.registeredComponents,
      builderBlock: o(),
      includeBlockProps: ((R = s == null ? void 0 : s()) == null ? void 0 : R.noWrap) === !0,
      isInteractive: !((_ = s == null ? void 0 : s()) != null && _.isRSC && lt === "rsc")
    };
  }
  return ye(() => {
    const P = o().id, O = o().animations;
    O && P && wo(
      O.map((R) => ({
        ...R,
        elementId: P
      }))
    );
  }, []), ye(() => {
    n.update = !0;
  }), /* @__PURE__ */ T(X, { children: f() ? /* @__PURE__ */ it(X, { children: [
    /* @__PURE__ */ T(Mo, { block: o(), context: e.context }),
    (I = s == null ? void 0 : s()) != null && I.noWrap ? t() ? /* @__PURE__ */ T(X, { children: (V = t()) == null ? void 0 : V.map((P, O) => /* @__PURE__ */ T(
      Vr,
      {
        repeatContext: P.context,
        block: P.block,
        registeredComponents: e.registeredComponents,
        linkComponent: e.linkComponent
      },
      Gr(O)
    )) }) : /* @__PURE__ */ T(
      $r,
      {
        componentRef: v().componentRef,
        componentOptions: v().componentOptions,
        blockChildren: v().blockChildren,
        context: v().context,
        registeredComponents: v().registeredComponents,
        linkComponent: v().linkComponent,
        builderBlock: v().builderBlock,
        includeBlockProps: v().includeBlockProps,
        isInteractive: v().isInteractive
      }
    ) : /* @__PURE__ */ T(X, { children: t() ? /* @__PURE__ */ T(X, { children: (W = t()) == null ? void 0 : W.map((P, O) => /* @__PURE__ */ T(
      Vr,
      {
        repeatContext: P.context,
        block: P.block,
        registeredComponents: e.registeredComponents,
        linkComponent: e.linkComponent
      },
      Gr(O)
    )) }) : /* @__PURE__ */ it(
      Wo,
      {
        Wrapper: a(),
        block: o(),
        context: e.context,
        children: [
          /* @__PURE__ */ T(
            $r,
            {
              componentRef: v().componentRef,
              componentOptions: v().componentOptions,
              blockChildren: v().blockChildren,
              context: v().context,
              registeredComponents: v().registeredComponents,
              linkComponent: v().linkComponent,
              builderBlock: v().builderBlock,
              includeBlockProps: v().includeBlockProps,
              isInteractive: v().isInteractive
            }
          ),
          (A = d()) == null ? void 0 : A.map((P) => /* @__PURE__ */ T(
            ht,
            {
              block: P,
              registeredComponents: e.registeredComponents,
              linkComponent: e.linkComponent,
              context: e.context
            },
            P.id
          ))
        ]
      }
    ) })
  ] }) : null });
}
function Xo(e) {
  const t = On(null);
  Ce(() => !1);
  function n() {
    var a;
    return [
      "builder-blocks",
      (a = e.blocks) != null && a.length ? "" : "no-blocks",
      e.classNameProp
    ].filter(Boolean).join(" ");
  }
  function r() {
    if (!e.path)
      return;
    const a = "this.", f = "component.options.";
    return e.path.startsWith(a) ? e.path.replace(a, "") : e.path.startsWith(f) ? e.path : `${f}${e.path || ""}`;
  }
  function o() {
    var a, f;
    Lt() && !((a = e.blocks) != null && a.length) && ((f = window.parent) == null || f.postMessage(
      {
        type: "builder.clickEmptyBlocks",
        data: {
          parentElementId: e.parent,
          dataPath: r()
        }
      },
      "*"
    ));
  }
  function s() {
    var a, f;
    Lt() && !((a = e.blocks) != null && a.length) && ((f = window.parent) == null || f.postMessage(
      {
        type: "builder.hoverEmptyBlocks",
        data: {
          parentElementId: e.parent,
          dataPath: r()
        }
      },
      "*"
    ));
  }
  return ye(() => {
  }, []), ye(() => {
  }, [e.blocks]), /* @__PURE__ */ it(X, { children: [
    /* @__PURE__ */ T("style", { children: `.props-blocks-wrapper-b46bf284 {
  display: flex;
  flex-direction: column;
  align-items: stretch;
}` }),
    /* @__PURE__ */ T(
      e.BlocksWrapper,
      {
        ref: t,
        className: n() + " props-blocks-wrapper-b46bf284",
        "builder-path": r(),
        "builder-parent-id": e.parent,
        style: e.styleProp,
        onClick: (a) => o(),
        onMouseEnter: (a) => s(),
        onKeyPress: (a) => o(),
        ...e.BlocksWrapperProps,
        children: e.children
      }
    )
  ] });
}
function Ho(e) {
  var r, o, s;
  const t = Mr(Xr), n = Mr(Pn);
  return /* @__PURE__ */ it(
    Xo,
    {
      blocks: e.blocks,
      parent: e.parent,
      path: e.path,
      styleProp: e.styleProp,
      BlocksWrapperProps: e.BlocksWrapperProps || ((r = e.context) == null ? void 0 : r.BlocksWrapperProps) || (t == null ? void 0 : t.BlocksWrapperProps),
      classNameProp: e.className,
      BlocksWrapper: ((o = e.context) == null ? void 0 : o.BlocksWrapper) || (t == null ? void 0 : t.BlocksWrapper),
      children: [
        e.children,
        e.blocks ? /* @__PURE__ */ T(X, { children: (s = e.blocks) == null ? void 0 : s.map((a) => /* @__PURE__ */ T(
          ht,
          {
            block: a,
            linkComponent: e.linkComponent,
            context: e.context || t,
            registeredComponents: e.registeredComponents || (n == null ? void 0 : n.registeredComponents)
          },
          a.id
        )) }) : null
      ]
    }
  );
}
const qo = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  default: Ho
}, Symbol.toStringTag, { value: "Module" }));
export {
  Ho as Blocks,
  Pn as ComponentsContext,
  Bo as InlinedStyles,
  qo as blocks,
  ft as evaluate,
  en as fastClone,
  bn as get,
  Ro as getSizesForBreakpoints,
  jo as mapStyleObjToStrIfNeeded,
  po as set,
  So as triggerAnimation
};
