// @builder.io/sdk-react/lib/edge/blocks-exports.mjs
"use client";
import { setClientUserAttributes as C } from "./server-entry-fb5f2b80.js";
import { lazy as t, createContext as e } from "react";
const s = t(() => import("./button-a478ff1a.js")), r = t(() => import("./columns-0b3181f1.js")), i = t(() => import("./fragment-1492bc7f.js")), a = t(() => import("./image-209835f4.js")), c = t(() => import("./section-d83e2dd3.js")), m = t(() => import("./symbol-10cb2f77.js")), l = t(() => import("./text-5915f038.js")), p = t(() => import("./video-52df64e2.js")), u = t(() => import("./blocks-0269007e.js").then((o) => o.blocks)), d = t(() => import("./content-variants-0dd65b50.js").then((o) => o.contentVariants)), x = e({
  content: null,
  context: {},
  localState: void 0,
  rootSetState() {
  },
  rootState: {},
  apiKey: null,
  apiVersion: void 0,
  componentInfos: {},
  inheritedStyles: {},
  BlocksWrapper: "div",
  BlocksWrapperProps: {},
  nonce: "",
  model: ""
});
export {
  u as Blocks,
  x as BuilderContext,
  s as Button,
  r as Columns,
  d as Content,
  i as Fragment,
  a as Image,
  c as Section,
  m as Symbol,
  l as Text,
  p as Video,
  C as setClientUserAttributes
};
