// @builder.io/sdk-react/lib/edge/columns-0b3181f1.js
import { jsxs as g, Fragment as $, jsx as r } from "react/jsx-runtime";
import { InlinedStyles as B, Blocks as L, mapStyleObjToStrIfNeeded as W, getSizesForBreakpoints as z } from "./blocks-0269007e.js";
import { DynamicRenderer as A } from "./dynamic-renderer-8b343325.js";
import { getClassPropName as N } from "./get-class-prop-name-770bdff2.js";
import "react";
import "./blocks-exports.mjs";
import "./server-entry-fb5f2b80.js";
const D = (e) => `builder-columns ${e}-breakpoints`;
function O(e) {
  var b, f;
  function c() {
    return typeof e.space == "number" ? e.space || 0 : 20;
  }
  function o() {
    return e.columns || [];
  }
  function s() {
    return e.stackColumnsAt || "tablet";
  }
  function y(t) {
    return t.link ? e.builderLinkComponent || "a" : "div";
  }
  function S(t) {
    var n;
    return ((n = o()[t]) == null ? void 0 : n.width) || 100 / o().length;
  }
  function x(t) {
    const n = S(t), i = c() * (o().length - 1) * (n / 100);
    return `calc(${n}% - ${i}px)`;
  }
  function a({
    stackedStyle: t,
    desktopStyle: n
  }) {
    return s() === "tablet" ? t : n;
  }
  function m({
    stackedStyle: t,
    desktopStyle: n
  }) {
    return s() === "never" ? n : t;
  }
  function u() {
    return e.stackColumnsAt === "never" ? "row" : e.reverseColumnsWhenStacked ? "column-reverse" : "column";
  }
  function C() {
    return {
      "--flex-dir": u(),
      "--flex-dir-tablet": a({
        stackedStyle: u(),
        desktopStyle: "row"
      })
    };
  }
  function v(t) {
    const n = t === 0 ? 0 : c(), i = x(t), l = `${n}px`, k = "100%", h = 0;
    return {
      ...{
        display: "flex",
        flexDirection: "column",
        alignItems: "stretch"
      },
      width: i,
      ["marginLeft"]: l,
      "--column-width-mobile": m({
        stackedStyle: k,
        desktopStyle: i
      }),
      "--column-margin-left-mobile": m({
        stackedStyle: h,
        desktopStyle: l
      }),
      "--column-width-tablet": a({
        stackedStyle: k,
        desktopStyle: i
      }),
      "--column-margin-left-tablet": a({
        stackedStyle: h,
        desktopStyle: l
      })
    };
  }
  function d(t) {
    var i, l;
    return z(
      ((l = (i = e.builderContext.content) == null ? void 0 : i.meta) == null ? void 0 : l.breakpoints) || {}
    )[t].max;
  }
  function w() {
    const t = `.${e.builderBlock.id}-breakpoints > .builder-column`;
    return `
        @media (max-width: ${d("medium")}px) {
          .${e.builderBlock.id}-breakpoints {
            flex-direction: var(--flex-dir-tablet);
            align-items: stretch;
          }

          ${t} {
            width: var(--column-width-tablet) !important;
            margin-left: var(--column-margin-left-tablet) !important;
          }
        }

        @media (max-width: ${d("small")}px) {
          .${e.builderBlock.id}-breakpoints {
            flex-direction: var(--flex-dir);
            align-items: stretch;
          }

          ${t} {
            width: var(--column-width-mobile) !important;
            margin-left: var(--column-margin-left-mobile) !important;
          }
        },
      `;
  }
  function p(t, n) {
    return {
      ...t.link ? {
        href: t.link
      } : {},
      [N()]: "builder-column",
      style: W(v(n))
    };
  }
  return /* @__PURE__ */ g($, { children: [
    /* @__PURE__ */ r("style", { children: `.div-0369db37 {
  display: flex;
  line-height: normal;
  height: 100%;
}` }),
    /* @__PURE__ */ g(
      "div",
      {
        className: D((b = e.builderBlock) == null ? void 0 : b.id) + " div-0369db37",
        style: C(),
        children: [
          /* @__PURE__ */ r(
            B,
            {
              id: "builderio-columns",
              styles: w(),
              nonce: e.builderContext.nonce
            }
          ),
          (f = e.columns) == null ? void 0 : f.map((t, n) => /* @__PURE__ */ r(
            A,
            {
              TagName: y(t),
              actionAttributes: {},
              attributes: p(t, n),
              children: /* @__PURE__ */ r(
                L,
                {
                  path: `columns.${n}.blocks`,
                  parent: e.builderBlock.id,
                  context: e.builderContext,
                  registeredComponents: e.builderComponents,
                  linkComponent: e.builderLinkComponent,
                  blocks: t.blocks,
                  styleProp: {
                    flexGrow: "1"
                  }
                }
              )
            },
            n
          ))
        ]
      }
    )
  ] });
}
export {
  O as default
};
