// @builder.io/sdk-react/lib/edge/image-209835f4.js
import { jsxs as l, Fragment as a, jsx as t } from "react/jsx-runtime";
import { useEffect as v } from "react";
import { getSrcSet as d } from "./image.helpers-3e926d47.js";
function w(e) {
  var r, u, s, g, m, h;
  function c() {
    var f;
    const i = e.image || e.src;
    if (!i || // We can auto add srcset for cdn.builder.io and shopify
    // images, otherwise you can supply this prop manually
    !(typeof i == "string" && (i.match(/builder\.io/) || i.match(/cdn\.shopify\.com/))))
      return e.srcset;
    if (!e.noWebp) {
      if (e.srcset && ((f = e.image) != null && f.includes("builder.io/api/v1/image"))) {
        if (!e.srcset.includes(e.image.split("?")[0]))
          return console.debug("Removed given srcset"), d(i);
      } else if (e.image && !e.srcset)
        return d(i);
      return d(i);
    }
  }
  function o() {
    var n;
    return (n = c == null ? void 0 : c()) != null && n.match(/builder\.io/) && !e.noWebp ? c().replace(/\?/g, "?format=webp&") : "";
  }
  function b() {
    const n = {
      position: "absolute",
      height: "100%",
      width: "100%",
      left: "0px",
      top: "0px"
    };
    return e.aspectRatio ? n : void 0;
  }
  return v(() => {
  }, []), /* @__PURE__ */ l(a, { children: [
    /* @__PURE__ */ t("style", { children: `.img-3cc22d08 {
  opacity: 1;
  transition: opacity 0.2s ease-in-out;
}.div-3cc22d08 {
  width: 100%;
  pointer-events: none;
  font-size: 0;
}.div-3cc22d08-2 {
  display: flex;
  flex-direction: column;
  align-items: stretch;
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
}` }),
    /* @__PURE__ */ l(a, { children: [
      /* @__PURE__ */ l("picture", { children: [
        o() ? /* @__PURE__ */ t("source", { type: "image/webp", srcSet: o() }) : null,
        /* @__PURE__ */ t(
          "img",
          {
            loading: e.highPriority ? "eager" : "lazy",
            fetchPriority: e.highPriority ? "high" : "auto",
            alt: e.altText,
            title: e.title,
            role: e.altText ? void 0 : "presentation",
            style: {
              objectPosition: e.backgroundPosition || "center",
              objectFit: e.backgroundSize || "cover",
              ...b()
            },
            className: "builder-image" + (e.className ? " " + e.className : "") + " img-3cc22d08",
            src: e.image,
            srcSet: c(),
            sizes: e.sizes
          }
        )
      ] }),
      e.aspectRatio && !((u = (r = e.builderBlock) == null ? void 0 : r.children) != null && u.length && e.fitContent) ? /* @__PURE__ */ t(
        "div",
        {
          className: "builder-image-sizer div-3cc22d08",
          style: {
            paddingTop: e.aspectRatio * 100 + "%"
          }
        }
      ) : null,
      (g = (s = e.builderBlock) == null ? void 0 : s.children) != null && g.length && e.fitContent ? /* @__PURE__ */ t(a, { children: e.children }) : null,
      !e.fitContent && ((h = (m = e.builderBlock) == null ? void 0 : m.children) != null && h.length) ? /* @__PURE__ */ t("div", { className: "div-3cc22d08-2", children: e.children }) : null
    ] })
  ] });
}
export {
  w as default
};
