// @builder.io/sdk-react/lib/edge/personalization-container-64f3cd71.js
import { jsxs as v, jsx as l, Fragment as j } from "react/jsx-runtime";
import { useRef as Q, useState as c, useEffect as z } from "react";
import { Blocks as m, InlinedStyles as X } from "./blocks-0269007e.js";
import { getPersonalizationScript as J, getUpdateVisibilityStylesScript as M, checkShouldRenderVariants as Y, DEFAULT_INDEX as C, SDKS_REQUIRING_RESET_APPROACH as f, InlinedScript as h, filterWithCustomTargeting as Z, getBlocksToRender as p } from "./content-variants-0dd65b50.js";
import { userAttributesService as N, getDefaultCanTrack as tt, TARGET as b, isEditing as et, isPreviewing as it } from "./server-entry-fb5f2b80.js";
import { getClassPropName as O } from "./get-class-prop-name-770bdff2.js";
import "./blocks-exports.mjs";
import "./dynamic-renderer-8b343325.js";
function mt(t) {
  var $, y, E, R, P, V, I, A, T, U, w, D, W;
  const a = Q(null), [q, F] = c(
    () => N.getUserAttributes()
  ), [S, nt] = c(
    () => {
      var e, n, i;
      return J(
        t.variants,
        ((e = t.builderBlock) == null ? void 0 : e.id) || "none",
        (i = (n = t.builderContext) == null ? void 0 : n.rootState) == null ? void 0 : i.locale
      );
    }
  ), [G, rt] = c(
    () => {
      var e, n, i;
      return M(
        t.variants,
        ((e = t.builderBlock) == null ? void 0 : e.id) || "none",
        (i = (n = t.builderContext) == null ? void 0 : n.rootState) == null ? void 0 : i.locale
      );
    }
  ), [x, lt] = c(() => []), [B, at] = c(
    () => {
      var e;
      return Y(
        t.variants,
        tt((e = t.builderContext) == null ? void 0 : e.canTrack)
      );
    }
  ), [k, H] = c(() => !1);
  function L() {
    return {
      ...t.attributes,
      [O()]: `builder-personalization-container ${t.attributes[O()] || ""}`
    };
  }
  function g() {
    return (t.variants || []).filter((e) => {
      var n, i, d, r;
      return Z(
        {
          ...(i = (n = t.builderContext) == null ? void 0 : n.rootState) != null && i.locale ? {
            locale: (r = (d = t.builderContext) == null ? void 0 : d.rootState) == null ? void 0 : r.locale
          } : {},
          ...q
        },
        e.query,
        e.startDate,
        e.endDate
      );
    });
  }
  function o() {
    var e;
    return p({
      variants: t.variants,
      fallbackBlocks: (e = t.builderBlock) == null ? void 0 : e.children,
      isHydrated: k,
      filteredVariants: g(),
      previewingIndex: t.previewingIndex
    });
  }
  function K() {
    return (t.variants || []).map(
      (e, n) => {
        var i;
        return `div[data-variant-id="${(i = t.builderBlock) == null ? void 0 : i.id}-${n}"] { display: none !important; } `;
      }
    ).join("");
  }
  return z(() => {
    var n;
    H(!0);
    const e = N.subscribeOnUserAttributesChange(
      (i) => {
        F(i);
      },
      {
        fireImmediately: b === "qwik"
      }
    );
    if (!(et() || it())) {
      const i = g()[0];
      a.current && (a.current.dispatchEvent(
        new CustomEvent("builder.variantLoaded", {
          detail: {
            variant: i || C,
            content: (n = t.builderContext) == null ? void 0 : n.content
          },
          bubbles: !0
        })
      ), new IntersectionObserver((r) => {
        r.forEach((u) => {
          var s;
          u.isIntersecting && a.current && a.current.dispatchEvent(
            new CustomEvent("builder.variantDisplayed", {
              detail: {
                variant: i || C,
                content: (s = t.builderContext) == null ? void 0 : s.content
              },
              bubbles: !0
            })
          );
        });
      }).observe(a.current));
    }
    x.push(e);
  }, []), z(() => () => {
    x.forEach((e) => e());
  }, []), /* @__PURE__ */ v("div", { ref: a, ...L(), children: [
    k && f.includes(b) ? /* @__PURE__ */ l(
      m,
      {
        blocks: o().blocks,
        parent: ($ = t.builderBlock) == null ? void 0 : $.id,
        path: o().path,
        context: t.builderContext,
        registeredComponents: t.builderComponents,
        BlocksWrapperProps: {
          ...(y = t.builderContext) == null ? void 0 : y.BlocksWrapperProps,
          "data-variant-id": `${(E = t.builderBlock) == null ? void 0 : E.id}-${o().index}`
        }
      }
    ) : null,
    !k && f.includes(b) || !f.includes(b) ? /* @__PURE__ */ v(j, { children: [
      B ? /* @__PURE__ */ v(j, { children: [
        /* @__PURE__ */ l(
          X,
          {
            nonce: ((R = t.builderContext) == null ? void 0 : R.nonce) || "",
            styles: K(),
            id: `variants-styles-${(P = t.builderBlock) == null ? void 0 : P.id}`
          }
        ),
        /* @__PURE__ */ l(
          h,
          {
            nonce: ((V = t.builderContext) == null ? void 0 : V.nonce) || "",
            scriptStr: G,
            id: `variants-visibility-script-${(I = t.builderBlock) == null ? void 0 : I.id}`
          }
        ),
        (A = t.variants) == null ? void 0 : A.map((e, n) => {
          var i, d, r, u, s, _;
          return /* @__PURE__ */ l(
            m,
            {
              BlocksWrapperProps: {
                ...(i = t.builderContext) == null ? void 0 : i.BlocksWrapperProps,
                "aria-hidden": !0,
                hidden: !0,
                "data-variant-id": `${(d = t.builderBlock) == null ? void 0 : d.id}-${n}`
              },
              blocks: e.blocks,
              parent: (r = t.builderBlock) == null ? void 0 : r.id,
              path: `component.options.variants.${n}.blocks`,
              context: t.builderContext,
              registeredComponents: t.builderComponents,
              children: /* @__PURE__ */ l(
                h,
                {
                  nonce: ((u = t.builderContext) == null ? void 0 : u.nonce) || "",
                  scriptStr: S,
                  id: `variants-script-${(s = t.builderBlock) == null ? void 0 : s.id}-${n}`
                }
              )
            },
            `${(_ = t.builderBlock) == null ? void 0 : _.id}-${n}`
          );
        })
      ] }) : null,
      /* @__PURE__ */ l(
        m,
        {
          blocks: o().blocks,
          parent: (T = t.builderBlock) == null ? void 0 : T.id,
          path: o().path,
          context: t.builderContext,
          registeredComponents: t.builderComponents,
          BlocksWrapperProps: {
            ...(U = t.builderContext) == null ? void 0 : U.BlocksWrapperProps,
            "data-variant-id": `${(w = t.builderBlock) == null ? void 0 : w.id}-${o().index}`
          },
          children: B ? /* @__PURE__ */ l(
            h,
            {
              nonce: ((D = t.builderContext) == null ? void 0 : D.nonce) || "",
              scriptStr: S,
              id: `variants-script-${(W = t.builderBlock) == null ? void 0 : W.id}-${C}`
            }
          ) : null
        }
      )
    ] }) : null
  ] });
}
export {
  mt as default
};
