// @builder.io/sdk-react/lib/edge/tabs-f9110171.js
import { jsxs as d, jsx as i } from "react/jsx-runtime";
import { useState as m } from "react";
import { Blocks as c } from "./blocks-0269007e.js";
import "./blocks-exports.mjs";
import "./server-entry-fb5f2b80.js";
import "./get-class-prop-name-770bdff2.js";
import "./dynamic-renderer-8b343325.js";
function x(t) {
  var b;
  const [n, a] = m(
    () => t.defaultActiveTab ? t.defaultActiveTab - 1 : 0
  );
  function o(e) {
    return t.tabs && t.tabs[e].content;
  }
  function r(e) {
    return e === n ? t.activeTabStyle : {};
  }
  function u(e) {
    e === n && t.collapsible ? a(-1) : a(e);
  }
  return /* @__PURE__ */ d("div", { children: [
    /* @__PURE__ */ i(
      "div",
      {
        className: "builder-tabs-wrap",
        style: {
          display: "flex",
          flexDirection: "row",
          justifyContent: t.tabHeaderLayout || "flex-start",
          overflow: "auto"
        },
        children: (b = t.tabs) == null ? void 0 : b.map((e, l) => /* @__PURE__ */ i(
          "span",
          {
            className: `builder-tab-wrap ${n === l ? "builder-tab-active" : ""}`,
            style: r(l),
            onClick: (s) => u(l),
            children: /* @__PURE__ */ i(
              c,
              {
                parent: t.builderBlock.id,
                path: `tabs.${l}.label`,
                blocks: e.label,
                context: t.builderContext,
                registeredComponents: t.builderComponents,
                linkComponent: t.builderLinkComponent
              }
            )
          },
          l
        ))
      }
    ),
    o(n) ? /* @__PURE__ */ i("div", { children: /* @__PURE__ */ i(
      c,
      {
        parent: t.builderBlock.id,
        path: `tabs.${n}.content`,
        blocks: o(n),
        context: t.builderContext,
        registeredComponents: t.builderComponents,
        linkComponent: t.builderLinkComponent
      }
    ) }) : null
  ] });
}
export {
  x as default
};
